<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Andy Beck" />

<meta name="date" content="2021-11-01" />

<title>1000G_two_positions_ALL</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Local Sequence Context Influence</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/theandyb/LSCI">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">1000G_two_positions_ALL</h1>
<h4 class="author">Andy Beck</h4>
<h4 class="date">2021-11-01</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-05-04
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>LSCI/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20211101code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20211101)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20211101code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20211101)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomtheandybLSCItree5df5f06c6dcd2310a5d7ab57681987d4c23f9fb3targetblank5df5f06a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/theandyb/LSCI/tree/5df5f06c6dcd2310a5d7ab57681987d4c23f9fb3" target="_blank">5df5f06</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomtheandybLSCItree5df5f06c6dcd2310a5d7ab57681987d4c23f9fb3targetblank5df5f06a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/theandyb/LSCI/tree/5df5f06c6dcd2310a5d7ab57681987d4c23f9fb3" target="_blank">5df5f06</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/disqus.html
    Ignored:    code/random/

Untracked files:
    Untracked:  1000G_single_position_ALL-tikzDictionary
    Untracked:  code/two_pos_functions.R
    Untracked:  output/tikz/

Unstaged changes:
    Modified:   analysis/BRIDGES_two_positions.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/1000G_two_positions_ALL.Rmd</code>) and HTML
(<code>docs/1000G_two_positions_ALL.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/theandyb/LSCI/blob/5df5f06c6dcd2310a5d7ab57681987d4c23f9fb3/analysis/1000G_two_positions_ALL.Rmd" target="_blank">5df5f06</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-05-04
</td>
<td>
wflow_publish("analysis/1000G_two_positions_ALL.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/theandyb/LSCI/fb68fdf5107911fddbd86e60528a5c9612f3488b/docs/1000G_two_positions_ALL.html" target="_blank">fb68fdf</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-05-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/theandyb/LSCI/blob/a1fa023fd7ac0336eff85572b0291ad85fedea21/analysis/1000G_two_positions_ALL.Rmd" target="_blank">a1fa023</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-05-04
</td>
<td>
wflow_publish("analysis/1000G_two_positions_ALL.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/theandyb/LSCI/d206789c1741f96d849c980167670cda330a86fa/docs/1000G_two_positions_ALL.html" target="_blank">d206789</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-04-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/theandyb/LSCI/blob/8785a91e1d687fc9f5da45833be192b5d83de1e9/analysis/1000G_two_positions_ALL.Rmd" target="_blank">8785a91</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-04-29
</td>
<td>
wflow_publish("analysis/1000G_two_positions_ALL.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/theandyb/LSCI/46d5054caf0b1691ef18540963a6f5b3357dd47e/docs/1000G_two_positions_ALL.html" target="_blank">46d5054</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-04-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/theandyb/LSCI/blob/c69b535b8d9228d9e4eaff66a92c7185d2b7ac20/analysis/1000G_two_positions_ALL.Rmd" target="_blank">c69b535</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-04-28
</td>
<td>
wflow_publish("analysis/1000G_two_positions_ALL.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/theandyb/LSCI/c33ff652181aef45b799411a45ed8917457b3dbf/docs/1000G_two_positions_ALL.html" target="_blank">c33ff65</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-04-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/theandyb/LSCI/blob/a1e7a6f6606942454f980c778a819e9f3607a560/analysis/1000G_two_positions_ALL.Rmd" target="_blank">a1e7a6f</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-04-28
</td>
<td>
wflow_publish("analysis/1000G_two_positions_ALL.Rmd")
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/theandyb/LSCI/blob/2aaffb26a379239076a7a15c719eca81eb176983/analysis/1000G_two_positions_ALL.Rmd" target="_blank">2aaffb2</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-04-28
</td>
<td>
Update documentation 28April2022
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/theandyb/LSCI/8f1cb1a25f560e383d98dc8ab9180c403f2a90a5/docs/1000G_two_positions_ALL.html" target="_blank">8f1cb1a</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-04-21
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/theandyb/LSCI/blob/491010c615557062c0e24d60452f2377ab977d83/analysis/1000G_two_positions_ALL.Rmd" target="_blank">491010c</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-04-21
</td>
<td>
wflow_publish(c("analysis/1000G_single_position_ALL.Rmd",
"analysis/1000G_two_positions_ALL.Rmd"))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/theandyb/LSCI/86380a3ec1cd6fc51774bc22304e97f9f8ffabdb/docs/1000G_two_positions_ALL.html" target="_blank">86380a3</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-01-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/theandyb/LSCI/blob/cccdcf5149e96ceea9f2de6c00159952121a4faf/analysis/1000G_two_positions_ALL.Rmd" target="_blank">cccdcf5</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-01-26
</td>
<td>
wflow_publish(c("analysis/1000G_two_positions_ALL.Rmd"))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/theandyb/LSCI/4643ea1a35afece317dde7b808200afca76a8b0b/docs/1000G_two_positions_ALL.html" target="_blank">4643ea1</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-01-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/theandyb/LSCI/blob/b205f471dafecb45418f0a51fc6d76c3044ed8ad/analysis/1000G_two_positions_ALL.Rmd" target="_blank">b205f47</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-01-20
</td>
<td>
wflow_publish(c("analysis/1000G_two_positions_ALL.Rmd",
"analysis/1000G_two_postions_comp.Rmd"))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/theandyb/LSCI/bcee6104baeb7c80b102faea4a04ffd5d3896da3/docs/1000G_two_positions_ALL.html" target="_blank">bcee610</a>
</td>
<td>
Andy Beck
</td>
<td>
2021-11-09
</td>
<td>
update site to include disqus
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/theandyb/LSCI/0fdbed84700c561213f47ff18c869ae52cfa0dc6/docs/1000G_two_positions_ALL.html" target="_blank">0fdbed8</a>
</td>
<td>
Andy Beck
</td>
<td>
2021-11-08
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/theandyb/LSCI/blob/27276dba015f3a5d067b6577be8ceabeaa8eadfe/analysis/1000G_two_positions_ALL.Rmd" target="_blank">27276db</a>
</td>
<td>
Andy Beck
</td>
<td>
2021-11-08
</td>
<td>
Add subsection header
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/theandyb/LSCI/ac3b368b8972b63cf5fd3267a2573e29d47ab89c/docs/1000G_two_positions_ALL.html" target="_blank">ac3b368</a>
</td>
<td>
Andy Beck
</td>
<td>
2021-11-08
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/theandyb/LSCI/blob/d08318eb010d4ececca08d50141ee848d7583307/analysis/1000G_two_positions_ALL.Rmd" target="_blank">d08318e</a>
</td>
<td>
Andy Beck
</td>
<td>
2021-11-08
</td>
<td>
Add two position model results
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/theandyb/LSCI/14543105cc2f4cae27f3316b81cb9280b93eddb0/docs/1000G_two_positions_ALL.html" target="_blank">1454310</a>
</td>
<td>
Andy Beck
</td>
<td>
2021-11-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/theandyb/LSCI/blob/6842cad988a7466fab1ee8ce3ccc32c7a5985830/analysis/1000G_two_positions_ALL.Rmd" target="_blank">6842cad</a>
</td>
<td>
Andy Beck
</td>
<td>
2021-11-01
</td>
<td>
flesh out some pages
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/theandyb/LSCI/6842cad988a7466fab1ee8ce3ccc32c7a5985830/docs/1000G_two_positions_ALL.html" target="_blank">6842cad</a>
</td>
<td>
Andy Beck
</td>
<td>
2021-11-01
</td>
<td>
flesh out some pages
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="g-two-position-models" class="section level1">
<h1>1000G Two Position Models</h1>
<div id="preface-shared-code" class="section level2">
<h2>Preface: Shared Code</h2>
<pre class="r"><code>#&#39; Get contingency table for two positions
#&#39;
#&#39; @param population 1KGP superpopulation code
#&#39; @param subtype Basic mutation subtype
#&#39; @param p1 First relative position
#&#39; @param p2 Second relative position
#&#39; @param data_dir Location of count tables
#&#39; @return Count table
singleton_count_2pos &lt;- function(population, subtype, p1, p2,
                                 data_dir = &quot;/net/snowwhite/home/beckandy/research/1000G_LSCI/output/all_count_2_pos/&quot;){
  f_name &lt;- paste0(data_dir, population, &quot;/&quot;, subtype, &quot;_p&quot;, p1, &quot;_q&quot;, p2, &quot;.csv&quot;)
  df &lt;- read_csv(f_name, col_types = cols()) %&gt;%
    filter(singletons &gt; 0)
  df_tab &lt;- xtabs(singletons ~ p1 + p2, data = df)
  return(df_tab)
}

#&#39; Perform statistical test for singleton counts stratified by nucleotides at two flanking positions
#&#39; @param population 1KGP superpopulation code
#&#39; @param subtype Simple mutation subtype
#&#39; @param stat_func Function to get statistic for pair of positions
#&#39; @return Data.frame with statistics at each pair of positions
stat_pair_all &lt;- function(population, subtype, stat_func, r_start=1, ...){
  final &lt;- data.frame(p1 = numeric(),
                      p2 = numeric(),
                      statistic = numeric())
  for(i in c(-10:-1,r_start:9)){
    for(j in (i+1):10){
      if(j == 0) next
      if(j == 1 &amp; r_start &gt; 1) next
      stat_val &lt;- stat_func(population, subtype, i, j, ...)
      final &lt;- bind_rows(final, data.frame(p1 = i,
                                           p2 = j,
                                           statistic = stat_val))
    }
  }
  return(final)
}

#&#39; Heatmap plot for two-position statistics
#&#39; 
#&#39; @param population 1KGP super-population code
#&#39; @param subtype Simple mutation subtype
#&#39; @param statistic_function Function which returns data frame with statistic at pairs of positions
#&#39; @param plot_title Text to begin plot title
#&#39; @param fill_label Legend title
#&#39; @return Figure with statistic values at each pair of positions
heatmap_plot &lt;- function(population, subtype, 
                         statistic_function, plot_title, fill_label, r_start = 1, ...){
  df &lt;- stat_pair_all(population, subtype, statistic_function, r_start, ...)
  
  if(!str_starts(subtype, &quot;cpg&quot;)){
    subtype2 &lt;- str_replace(subtype, &quot;_&quot;, &quot; -&gt; &quot;)
  } else {
    subtype &lt;- str_sub(subtype, 5)
    subtype2 &lt;- paste0(&quot;(cpg) &quot;, str_replace(subtype, &quot;_&quot;, &quot; -&gt; &quot;))
  }
  
  p &lt;- df %&gt;%
    ggplot(aes(x = p2, y = p1, fill = statistic)) +
    geom_tile() +
    ggtitle(paste0(plot_title, subtype2),
            paste0(&quot;Population: &quot;, population,
                   &quot;; Min: &quot;, round(min(df$statistic), 2), 
                   &quot;; Max: &quot;, round(max(df$statistic), 2)))+
    xlab(&quot;Relative Position 2&quot;) +
    ylab(&quot;Relative Position 1&quot;) +
    labs(fill = fill_label) +
    scale_fill_distiller(palette = &quot;Reds&quot;, direction = 1)
  return(p)
}</code></pre>
</div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>In a <a href="1000G_single_position_ALL.html">previous document</a>
we evaluated the influence of individual positions on the mutation rate
as a function of relative position. In do this we marginalized over the
nucleotides at other positions, ignoring any potential interaction
effect. Here in this document we will being to explore the extent to
which interactions among flanking positions influence the rate of
substitution at a focal site by first focusing on two-way
interactions.</p>
<p>Challenges we face here are:</p>
<ol style="list-style-type: decimal">
<li>We know that the marginal distributions at single positions in our
control samples (and the genome-wide background rate) don’t match
corresponding marginals in the singletons</li>
<li>When considering the 2-way contingency table at each pair of
positions, the row and column marginals don’t predict the observed 4x4
cells in the genome-wide and control counts (i.e. the nucleotides at the
two positions are not independent)</li>
</ol>
</div>
<div id="chi-square-test-of-independence" class="section level2">
<h2>Chi-Square Test of Independence</h2>
<p>For this analysis we’ll be looking at only our singleton counts. For
each sub type at each pair of positions we observe a 4x4 2-way
contingency table. Performing a chi square test of independence asks if
we have evidence to reject the null hypothesis that the two “variables”
(sites) are independent; i.e. the number of times we see AA is the
product of the row marginal for A and the column marginal for A,
etc.</p>
<p>This may seem like a silly thing to do since we already know
genome-wide the rate of di-nucleotides is not predicted by the product
of the frequency of the two nucleotides. It is in fact the case that
this pattern is also observed in the singleton counts, and nothing here
is probably worth mentioning in the paper. But anyways, here are those
results.</p>
<pre class="r"><code>#&#39; Perform chi-sq test of independence for singleton counts stratified by nucleotides at two flanking positions
#&#39; @param population 1KGP superpopulation code
#&#39; @param subtype Simple mutation subtype
#&#39; @return Chi square statistic
chi_sq_ind &lt;- function(population, subtype, p1, p2){
  df_tab &lt;- singleton_count_2pos(population, subtype, p1, p2)
  stat_val &lt;- (chisq.test(df_tab)$statistic %&gt;% unname())
  return(stat_val)
}

chi_sq_ind_plot &lt;- function(population, subtype){
  p &lt;- heatmap_plot(population, subtype, chi_sq_ind, &quot;Chi Sq Test of Independence: &quot;, &quot;Chi Sq Statistic&quot;)
  return(p)
}</code></pre>
<div id="at_cg" class="section level3">
<h3>AT_CG</h3>
<p><img src="figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-1">
Past versions of unnamed-chunk-4-1.png
</button>
</p>
<div id="fig-unnamed-chunk-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/fb68fdf5107911fddbd86e60528a5c9612f3488b/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-4-1.png" target="_blank">fb68fdf</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-05-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>We see here the expected pattern that directly adjacent
di-nucloetides are not observed at frequencies suggested by the product
of their marginal frequencies, even when looking only at those seen
flanking a singleton. Might I eventually want to look at a version of
this that isn’t heavily affected by the large sample size? Maybe.</p>
</div>
<div id="at_gc" class="section level3">
<h3>AT_GC</h3>
<p><img src="figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/fb68fdf5107911fddbd86e60528a5c9612f3488b/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-5-1.png" target="_blank">fb68fdf</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-05-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/4643ea1a35afece317dde7b808200afca76a8b0b/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-5-1.png" target="_blank">4643ea1</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-01-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/ac3b368b8972b63cf5fd3267a2573e29d47ab89c/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-5-1.png" target="_blank">ac3b368</a>
</td>
<td>
Andy Beck
</td>
<td>
2021-11-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="at_ta" class="section level3">
<h3>AT_TA</h3>
<p><img src="figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-1">
Past versions of unnamed-chunk-6-1.png
</button>
</p>
<div id="fig-unnamed-chunk-6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/fb68fdf5107911fddbd86e60528a5c9612f3488b/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-6-1.png" target="_blank">fb68fdf</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-05-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/4643ea1a35afece317dde7b808200afca76a8b0b/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-6-1.png" target="_blank">4643ea1</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-01-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/ac3b368b8972b63cf5fd3267a2573e29d47ab89c/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-6-1.png" target="_blank">ac3b368</a>
</td>
<td>
Andy Beck
</td>
<td>
2021-11-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="gc_at" class="section level3">
<h3>GC_AT</h3>
<p><img src="figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-1">
Past versions of unnamed-chunk-7-1.png
</button>
</p>
<div id="fig-unnamed-chunk-7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/fb68fdf5107911fddbd86e60528a5c9612f3488b/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-7-1.png" target="_blank">fb68fdf</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-05-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/4643ea1a35afece317dde7b808200afca76a8b0b/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-7-1.png" target="_blank">4643ea1</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-01-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/ac3b368b8972b63cf5fd3267a2573e29d47ab89c/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-7-1.png" target="_blank">ac3b368</a>
</td>
<td>
Andy Beck
</td>
<td>
2021-11-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cpg_gc_at" class="section level3">
<h3>cpg_GC_AT</h3>
<p><img src="figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-1">
Past versions of unnamed-chunk-8-1.png
</button>
</p>
<div id="fig-unnamed-chunk-8-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/fb68fdf5107911fddbd86e60528a5c9612f3488b/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-8-1.png" target="_blank">fb68fdf</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-05-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/4643ea1a35afece317dde7b808200afca76a8b0b/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-8-1.png" target="_blank">4643ea1</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-01-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/ac3b368b8972b63cf5fd3267a2573e29d47ab89c/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-8-1.png" target="_blank">ac3b368</a>
</td>
<td>
Andy Beck
</td>
<td>
2021-11-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="gc_ta" class="section level3">
<h3>GC_TA</h3>
<p><img src="figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-1">
Past versions of unnamed-chunk-9-1.png
</button>
</p>
<div id="fig-unnamed-chunk-9-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/fb68fdf5107911fddbd86e60528a5c9612f3488b/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-9-1.png" target="_blank">fb68fdf</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-05-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/4643ea1a35afece317dde7b808200afca76a8b0b/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-9-1.png" target="_blank">4643ea1</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-01-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/ac3b368b8972b63cf5fd3267a2573e29d47ab89c/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-9-1.png" target="_blank">ac3b368</a>
</td>
<td>
Andy Beck
</td>
<td>
2021-11-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cpg_gc_ta" class="section level3">
<h3>cpg_GC_TA</h3>
<p><img src="figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-10-1">
Past versions of unnamed-chunk-10-1.png
</button>
</p>
<div id="fig-unnamed-chunk-10-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/fb68fdf5107911fddbd86e60528a5c9612f3488b/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-10-1.png" target="_blank">fb68fdf</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-05-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/4643ea1a35afece317dde7b808200afca76a8b0b/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-10-1.png" target="_blank">4643ea1</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-01-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/ac3b368b8972b63cf5fd3267a2573e29d47ab89c/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-10-1.png" target="_blank">ac3b368</a>
</td>
<td>
Andy Beck
</td>
<td>
2021-11-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="goodness-of-fit-estimate-distribution-with-controls"
class="section level2">
<h2>Goodness of Fit: Estimate Distribution with Controls</h2>
<p>The idea here is to estimate the frequency of flanking dimers using
the dimers we observe flanking control positions. This ignores the fact
that the distribution of individual flanking nucleotides has already
been shown to differ between the singletons and the matched controls;
what I think we want is a way to test for an interaction while
accounting for the different individual marginals.</p>
<p>But anyways, here’s the code to get the chi square GOF statistic for
a given sub type at a pair of positions:</p>
<pre class="r"><code>chi_sq_gof_pair &lt;- function(population, subtype, p1, p2) {
  data_dir &lt;- &quot;/net/snowwhite/home/beckandy/research/1000G_LSCI/output/all_count_2_pos/&quot;
  f_name &lt;- paste0(data_dir, population, &quot;/&quot;, subtype, &quot;_p&quot;, p1, &quot;_q&quot;, p2, &quot;.csv&quot;)
  df &lt;- read_csv(f_name, col_types = cols()) %&gt;%
    filter(singletons &gt; 0)
  df$p_c &lt;- df$controls / sum(df$controls)
  df$e_s &lt;- sum(df$singletons) * df$p_c
  df$chi_res &lt;- (df$e_s - df$singletons)^2 / df$e_s
  df$p_s &lt;- df$singletons / sum(df$singletons)
  df$kl_res &lt;- df$p_s * log(df$p_s / df$p_c)
  return(sum(df$chi_res))
}

chi_sq_re_pair &lt;- function(population, subtype, p1, p2){
  data_dir &lt;- &quot;/net/snowwhite/home/beckandy/research/1000G_LSCI/output/all_count_2_pos/&quot;
  f_name &lt;- paste0(data_dir, population, &quot;/&quot;, subtype, &quot;_p&quot;, p1, &quot;_q&quot;, p2, &quot;.csv&quot;)
  df &lt;- read_csv(f_name, col_types = cols()) %&gt;%
    filter(singletons &gt; 0) %&gt;%
    mutate(p_c = controls / sum(controls),
           p_s = singletons / sum(singletons)) %&gt;%
    mutate(re = p_s * log(p_s / p_c))
  return(sum(df$re))
}</code></pre>
<pre class="r"><code>chi_sq_gof_plot &lt;- function(population, subtype){
  p &lt;- heatmap_plot(population, subtype, chi_sq_gof_pair, 
                    &quot;Control Di-Nucleotide Rate: &quot;, &quot;Chi Sq Statistic&quot;)
  return(p)
}

chi_sq_re_plot &lt;- function(population, subtype){
  p &lt;- heatmap_plot(population, subtype, chi_sq_re_pair, 
                    &quot;Control Di-Nucleotide Rate: &quot;, &quot;Relative Entropy&quot;)
  return(p)
}</code></pre>
<div id="at_cg-1" class="section level3">
<h3>AT_CG</h3>
<pre class="r"><code>subtype &lt;- &quot;AT_CG&quot;
p &lt;- chi_sq_re_plot(&quot;ALL&quot;, subtype)
print(p)</code></pre>
<p><img src="figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-13-1">
Past versions of unnamed-chunk-13-1.png
</button>
</p>
<div id="fig-unnamed-chunk-13-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/fb68fdf5107911fddbd86e60528a5c9612f3488b/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-13-1.png" target="_blank">fb68fdf</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-05-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/8f1cb1a25f560e383d98dc8ab9180c403f2a90a5/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-13-1.png" target="_blank">8f1cb1a</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-04-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/ac3b368b8972b63cf5fd3267a2573e29d47ab89c/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-13-1.png" target="_blank">ac3b368</a>
</td>
<td>
Andy Beck
</td>
<td>
2021-11-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="at_gc-1" class="section level3">
<h3>AT_GC</h3>
<pre class="r"><code>subtype &lt;- &quot;AT_GC&quot;
p &lt;- chi_sq_re_plot(&quot;ALL&quot;, subtype)
print(p)</code></pre>
<p><img src="figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-1">
Past versions of unnamed-chunk-14-1.png
</button>
</p>
<div id="fig-unnamed-chunk-14-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/fb68fdf5107911fddbd86e60528a5c9612f3488b/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-14-1.png" target="_blank">fb68fdf</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-05-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/8f1cb1a25f560e383d98dc8ab9180c403f2a90a5/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-14-1.png" target="_blank">8f1cb1a</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-04-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/ac3b368b8972b63cf5fd3267a2573e29d47ab89c/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-14-1.png" target="_blank">ac3b368</a>
</td>
<td>
Andy Beck
</td>
<td>
2021-11-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="at_ta-1" class="section level3">
<h3>AT_TA</h3>
<pre class="r"><code>subtype &lt;- &quot;AT_TA&quot;
p &lt;- chi_sq_re_plot(&quot;ALL&quot;, subtype)
print(p)</code></pre>
<p><img src="figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-15-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-15-1">
Past versions of unnamed-chunk-15-1.png
</button>
</p>
<div id="fig-unnamed-chunk-15-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/fb68fdf5107911fddbd86e60528a5c9612f3488b/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-15-1.png" target="_blank">fb68fdf</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-05-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/d206789c1741f96d849c980167670cda330a86fa/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-15-1.png" target="_blank">d206789</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-04-29
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/46d5054caf0b1691ef18540963a6f5b3357dd47e/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-15-1.png" target="_blank">46d5054</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-04-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/8f1cb1a25f560e383d98dc8ab9180c403f2a90a5/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-15-1.png" target="_blank">8f1cb1a</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-04-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/ac3b368b8972b63cf5fd3267a2573e29d47ab89c/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-15-1.png" target="_blank">ac3b368</a>
</td>
<td>
Andy Beck
</td>
<td>
2021-11-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="gc_at-1" class="section level3">
<h3>GC_AT</h3>
<pre class="r"><code>subtype &lt;- &quot;GC_AT&quot;
p &lt;- chi_sq_re_plot(&quot;ALL&quot;, subtype)
print(p)</code></pre>
<p><img src="figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-16-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-16-1">
Past versions of unnamed-chunk-16-1.png
</button>
</p>
<div id="fig-unnamed-chunk-16-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/fb68fdf5107911fddbd86e60528a5c9612f3488b/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-16-1.png" target="_blank">fb68fdf</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-05-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/d206789c1741f96d849c980167670cda330a86fa/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-16-1.png" target="_blank">d206789</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-04-29
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/46d5054caf0b1691ef18540963a6f5b3357dd47e/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-16-1.png" target="_blank">46d5054</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-04-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/8f1cb1a25f560e383d98dc8ab9180c403f2a90a5/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-16-1.png" target="_blank">8f1cb1a</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-04-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/ac3b368b8972b63cf5fd3267a2573e29d47ab89c/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-16-1.png" target="_blank">ac3b368</a>
</td>
<td>
Andy Beck
</td>
<td>
2021-11-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cpg_gc_at-1" class="section level3">
<h3>cpg_GC_AT</h3>
<pre class="r"><code>subtype &lt;- &quot;cpg_GC_AT&quot;
p &lt;- chi_sq_re_plot(&quot;ALL&quot;, subtype)
print(p)</code></pre>
<p><img src="figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-17-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-17-1">
Past versions of unnamed-chunk-17-1.png
</button>
</p>
<div id="fig-unnamed-chunk-17-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/fb68fdf5107911fddbd86e60528a5c9612f3488b/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-17-1.png" target="_blank">fb68fdf</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-05-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/46d5054caf0b1691ef18540963a6f5b3357dd47e/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-17-1.png" target="_blank">46d5054</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-04-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/8f1cb1a25f560e383d98dc8ab9180c403f2a90a5/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-17-1.png" target="_blank">8f1cb1a</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-04-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/ac3b368b8972b63cf5fd3267a2573e29d47ab89c/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-17-1.png" target="_blank">ac3b368</a>
</td>
<td>
Andy Beck
</td>
<td>
2021-11-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="gc_ta-1" class="section level3">
<h3>GC_TA</h3>
<pre class="r"><code>subtype &lt;- &quot;GC_TA&quot;
p &lt;- chi_sq_re_plot(&quot;ALL&quot;, subtype)
print(p)</code></pre>
<p><img src="figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-18-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-18-1">
Past versions of unnamed-chunk-18-1.png
</button>
</p>
<div id="fig-unnamed-chunk-18-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/fb68fdf5107911fddbd86e60528a5c9612f3488b/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-18-1.png" target="_blank">fb68fdf</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-05-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/46d5054caf0b1691ef18540963a6f5b3357dd47e/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-18-1.png" target="_blank">46d5054</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-04-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/8f1cb1a25f560e383d98dc8ab9180c403f2a90a5/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-18-1.png" target="_blank">8f1cb1a</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-04-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/ac3b368b8972b63cf5fd3267a2573e29d47ab89c/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-18-1.png" target="_blank">ac3b368</a>
</td>
<td>
Andy Beck
</td>
<td>
2021-11-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cpg_gc_ta-1" class="section level3">
<h3>cpg_GC_TA</h3>
<pre class="r"><code>subtype &lt;- &quot;cpg_GC_TA&quot;
p &lt;- chi_sq_re_plot(&quot;ALL&quot;, subtype)
print(p)</code></pre>
<p><img src="figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-19-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-19-1">
Past versions of unnamed-chunk-19-1.png
</button>
</p>
<div id="fig-unnamed-chunk-19-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/fb68fdf5107911fddbd86e60528a5c9612f3488b/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-19-1.png" target="_blank">fb68fdf</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-05-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/46d5054caf0b1691ef18540963a6f5b3357dd47e/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-19-1.png" target="_blank">46d5054</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-04-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/8f1cb1a25f560e383d98dc8ab9180c403f2a90a5/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-19-1.png" target="_blank">8f1cb1a</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-04-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/ac3b368b8972b63cf5fd3267a2573e29d47ab89c/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-19-1.png" target="_blank">ac3b368</a>
</td>
<td>
Andy Beck
</td>
<td>
2021-11-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="gc_cg" class="section level3">
<h3>GC_CG</h3>
<pre class="r"><code>subtype &lt;- &quot;GC_CG&quot;
p &lt;- chi_sq_re_plot(&quot;ALL&quot;, subtype)
print(p)</code></pre>
<p><img src="figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-20-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-20-1">
Past versions of unnamed-chunk-20-1.png
</button>
</p>
<div id="fig-unnamed-chunk-20-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/fb68fdf5107911fddbd86e60528a5c9612f3488b/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-20-1.png" target="_blank">fb68fdf</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-05-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/46d5054caf0b1691ef18540963a6f5b3357dd47e/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-20-1.png" target="_blank">46d5054</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-04-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/8f1cb1a25f560e383d98dc8ab9180c403f2a90a5/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-20-1.png" target="_blank">8f1cb1a</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-04-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/ac3b368b8972b63cf5fd3267a2573e29d47ab89c/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-20-1.png" target="_blank">ac3b368</a>
</td>
<td>
Andy Beck
</td>
<td>
2021-11-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cpg_gc_cg" class="section level3">
<h3>cpg_GC_CG</h3>
<pre class="r"><code>subtype &lt;- &quot;cpg_GC_CG&quot;
p &lt;- chi_sq_re_plot(&quot;ALL&quot;, subtype)
print(p)</code></pre>
<p><img src="figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-21-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-21-1">
Past versions of unnamed-chunk-21-1.png
</button>
</p>
<div id="fig-unnamed-chunk-21-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/fb68fdf5107911fddbd86e60528a5c9612f3488b/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-21-1.png" target="_blank">fb68fdf</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-05-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/8f1cb1a25f560e383d98dc8ab9180c403f2a90a5/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-21-1.png" target="_blank">8f1cb1a</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-04-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/ac3b368b8972b63cf5fd3267a2573e29d47ab89c/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-21-1.png" target="_blank">ac3b368</a>
</td>
<td>
Andy Beck
</td>
<td>
2021-11-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="gof-control-interaction-estimation" class="section level2">
<h2>GOF: Control interaction estimation</h2>
<p>An idea I want to try here is to try and estimate a shared
interaction parameter using only the control frequencies and apply this
to the singleton frequencies. In a nutshell, we can think of the 16
frequencies we see in the control table as being the product of two
marginal frequencies and an additional interaction term (question:
should this be additive or multiplicative?). In the multiplicative
formulation of the interaction, the frequency in each cell can be
written as:</p>
<p><span class="math display">\[
f^C_{i,j} = p^C_{i.}p^C_{.j}p_{ij}
\]</span></p>
<p>and the idea here would be to estimate the <span
class="math inline">\(p_{ij}\)</span> interaction terms using the
controls and using these to adjust the cell frequencies in the
singletons and then assess the fit using a <span
class="math inline">\(\chi^2\)</span> goodness of fit test.</p>
<p>Let’s try it out with the multiplicative version:</p>
<pre class="r"><code>get_mar &lt;- function(freq_tab){
  n_t &lt;- sum(freq_tab)
  col_mar &lt;- colSums(freq_tab) / n_t
  row_mar &lt;- rowSums(freq_tab) / n_t
  
  prod_mat &lt;- row_mar %*% t(col_mar)
  return(prod_mat)
}

get_int &lt;- function(freq_tab){
  n_t &lt;- sum(freq_tab)
  col_mar &lt;- colSums(freq_tab) / n_t
  row_mar &lt;- rowSums(freq_tab) / n_t
  
  prod_mat &lt;- row_mar %*% t(col_mar)
  int_mat &lt;- (freq_tab / n_t) / prod_mat
  return(int_mat)
}

control_int_mult_pair &lt;- function(population, subtype, p1, p2, type = &quot;gof&quot;) {
  data_dir &lt;- &quot;/net/snowwhite/home/beckandy/research/1000G_LSCI/output/all_count_2_pos/&quot;
  f_name &lt;- paste0(data_dir, population, &quot;/&quot;, subtype, &quot;_p&quot;, p1, &quot;_q&quot;, p2, &quot;.csv&quot;)
  df &lt;- read_csv(f_name, col_types = cols()) %&gt;%
    filter(singletons &gt; 0)
  control_tab &lt;- df %&gt;% 
    select(p1, p2, controls) %&gt;% 
    pivot_wider(names_from = p2, values_from = controls) %&gt;%
    remove_rownames() %&gt;%
    column_to_rownames(var = &#39;p1&#39;)
  singleton_tab &lt;- df %&gt;% 
    select(p1, p2, singletons) %&gt;% 
    pivot_wider(names_from = p2, values_from = singletons) %&gt;%
    remove_rownames() %&gt;%
    column_to_rownames(var = &#39;p1&#39;)

  exp_s &lt;- get_mar(singleton_tab) * get_int(control_tab)
  exp_p_s &lt;- exp_s / sum(exp_s) # normalize
  exp_s &lt;- exp_p_s * sum(singleton_tab)
  chi_sq &lt;- (exp_s - singleton_tab)^2 / exp_s
  
  if(type == &quot;gof&quot;){
    stat_val &lt;- sum(chi_sq)
  } else{
    p_s &lt;- singleton_tab / sum(singleton_tab)
    re &lt;- p_s * log(p_s / exp_p_s)
    stat_val &lt;- sum(re)
  }
  
  return(stat_val)
}

control_mult_int_plot &lt;- function(population, subtype, r_start = 1, type = &quot;gof&quot;){
  if(type == &quot;gof&quot;) {
    l_title &lt;- &quot;Chi Sq. Statistic&quot;
  } else {
    l_title &lt;- &quot;Relative Entropy&quot;
  }
  p &lt;- heatmap_plot(population, subtype, control_int_mult_pair, 
                    &quot;Control Estimate Interaction: &quot;,
                    l_title, r_start , type)
  return(p)
}</code></pre>
<div id="at_cg-2" class="section level3">
<h3>AT_CG</h3>
<pre class="r"><code>subtype &lt;- &quot;AT_GC&quot;
p &lt;- control_mult_int_plot(&quot;ALL&quot;, subtype, type = &quot;re&quot;)
print(p)</code></pre>
<p><img src="figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-23-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-23-1">
Past versions of unnamed-chunk-23-1.png
</button>
</p>
<div id="fig-unnamed-chunk-23-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/fb68fdf5107911fddbd86e60528a5c9612f3488b/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-23-1.png" target="_blank">fb68fdf</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-05-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/d206789c1741f96d849c980167670cda330a86fa/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-23-1.png" target="_blank">d206789</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-04-29
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/c33ff652181aef45b799411a45ed8917457b3dbf/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-23-1.png" target="_blank">c33ff65</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-04-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/8f1cb1a25f560e383d98dc8ab9180c403f2a90a5/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-23-1.png" target="_blank">8f1cb1a</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-04-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="at_gc-2" class="section level3">
<h3>AT_GC</h3>
<pre class="r"><code>subtype &lt;- &quot;AT_GC&quot;
p &lt;- control_mult_int_plot(&quot;ALL&quot;, subtype, type = &quot;re&quot;)
print(p)</code></pre>
<p><img src="figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-24-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-24-1">
Past versions of unnamed-chunk-24-1.png
</button>
</p>
<div id="fig-unnamed-chunk-24-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/fb68fdf5107911fddbd86e60528a5c9612f3488b/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-24-1.png" target="_blank">fb68fdf</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-05-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/d206789c1741f96d849c980167670cda330a86fa/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-24-1.png" target="_blank">d206789</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-04-29
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/c33ff652181aef45b799411a45ed8917457b3dbf/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-24-1.png" target="_blank">c33ff65</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-04-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/8f1cb1a25f560e383d98dc8ab9180c403f2a90a5/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-24-1.png" target="_blank">8f1cb1a</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-04-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="at_ta-2" class="section level3">
<h3>AT_TA</h3>
<pre class="r"><code>subtype &lt;- &quot;AT_TA&quot;
p &lt;- control_mult_int_plot(&quot;ALL&quot;, subtype, type = &quot;re&quot;)
print(p)</code></pre>
<p><img src="figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-25-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-25-1">
Past versions of unnamed-chunk-25-1.png
</button>
</p>
<div id="fig-unnamed-chunk-25-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/fb68fdf5107911fddbd86e60528a5c9612f3488b/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-25-1.png" target="_blank">fb68fdf</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-05-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/d206789c1741f96d849c980167670cda330a86fa/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-25-1.png" target="_blank">d206789</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-04-29
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/c33ff652181aef45b799411a45ed8917457b3dbf/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-25-1.png" target="_blank">c33ff65</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-04-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/8f1cb1a25f560e383d98dc8ab9180c403f2a90a5/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-25-1.png" target="_blank">8f1cb1a</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-04-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="gc_at-2" class="section level3">
<h3>GC_AT</h3>
<pre class="r"><code>subtype &lt;- &quot;GC_AT&quot;
p &lt;- control_mult_int_plot(&quot;ALL&quot;, subtype, type = &quot;re&quot;)
print(p)</code></pre>
<p><img src="figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-26-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-26-1">
Past versions of unnamed-chunk-26-1.png
</button>
</p>
<div id="fig-unnamed-chunk-26-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/fb68fdf5107911fddbd86e60528a5c9612f3488b/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-26-1.png" target="_blank">fb68fdf</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-05-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/d206789c1741f96d849c980167670cda330a86fa/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-26-1.png" target="_blank">d206789</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-04-29
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/c33ff652181aef45b799411a45ed8917457b3dbf/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-26-1.png" target="_blank">c33ff65</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-04-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/8f1cb1a25f560e383d98dc8ab9180c403f2a90a5/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-26-1.png" target="_blank">8f1cb1a</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-04-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cpg_gc_at-2" class="section level3">
<h3>cpg_GC_AT</h3>
<pre class="r"><code>subtype &lt;- &quot;cpg_GC_AT&quot;
p &lt;- control_mult_int_plot(&quot;ALL&quot;, subtype, type = &quot;re&quot;)
print(p)</code></pre>
<p><img src="figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-27-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-27-1">
Past versions of unnamed-chunk-27-1.png
</button>
</p>
<div id="fig-unnamed-chunk-27-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/fb68fdf5107911fddbd86e60528a5c9612f3488b/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-27-1.png" target="_blank">fb68fdf</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-05-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/d206789c1741f96d849c980167670cda330a86fa/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-27-1.png" target="_blank">d206789</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-04-29
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/c33ff652181aef45b799411a45ed8917457b3dbf/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-27-1.png" target="_blank">c33ff65</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-04-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/8f1cb1a25f560e383d98dc8ab9180c403f2a90a5/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-27-1.png" target="_blank">8f1cb1a</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-04-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="gc_ta-2" class="section level3">
<h3>GC_TA</h3>
<pre class="r"><code>subtype &lt;- &quot;GC_TA&quot;
p &lt;- control_mult_int_plot(&quot;ALL&quot;, subtype, type = &quot;re&quot;)
print(p)</code></pre>
<p><img src="figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-28-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-28-1">
Past versions of unnamed-chunk-28-1.png
</button>
</p>
<div id="fig-unnamed-chunk-28-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/fb68fdf5107911fddbd86e60528a5c9612f3488b/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-28-1.png" target="_blank">fb68fdf</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-05-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/d206789c1741f96d849c980167670cda330a86fa/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-28-1.png" target="_blank">d206789</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-04-29
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/c33ff652181aef45b799411a45ed8917457b3dbf/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-28-1.png" target="_blank">c33ff65</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-04-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/8f1cb1a25f560e383d98dc8ab9180c403f2a90a5/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-28-1.png" target="_blank">8f1cb1a</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-04-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cpg_gc_ta-2" class="section level3">
<h3>cpg_GC_TA</h3>
<pre class="r"><code>subtype &lt;- &quot;cpg_GC_TA&quot;
p &lt;- control_mult_int_plot(&quot;ALL&quot;, subtype, type = &quot;re&quot;)
print(p)</code></pre>
<p><img src="figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-29-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-29-1">
Past versions of unnamed-chunk-29-1.png
</button>
</p>
<div id="fig-unnamed-chunk-29-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/fb68fdf5107911fddbd86e60528a5c9612f3488b/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-29-1.png" target="_blank">fb68fdf</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-05-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/d206789c1741f96d849c980167670cda330a86fa/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-29-1.png" target="_blank">d206789</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-04-29
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/c33ff652181aef45b799411a45ed8917457b3dbf/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-29-1.png" target="_blank">c33ff65</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-04-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/8f1cb1a25f560e383d98dc8ab9180c403f2a90a5/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-29-1.png" target="_blank">8f1cb1a</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-04-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="gc_cg-1" class="section level3">
<h3>GC_CG</h3>
<pre class="r"><code>subtype &lt;- &quot;GC_CG&quot;
p &lt;- control_mult_int_plot(&quot;ALL&quot;, subtype, type = &quot;re&quot;)
print(p)</code></pre>
<p><img src="figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-30-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-30-1">
Past versions of unnamed-chunk-30-1.png
</button>
</p>
<div id="fig-unnamed-chunk-30-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/fb68fdf5107911fddbd86e60528a5c9612f3488b/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-30-1.png" target="_blank">fb68fdf</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-05-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/d206789c1741f96d849c980167670cda330a86fa/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-30-1.png" target="_blank">d206789</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-04-29
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/c33ff652181aef45b799411a45ed8917457b3dbf/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-30-1.png" target="_blank">c33ff65</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-04-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/8f1cb1a25f560e383d98dc8ab9180c403f2a90a5/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-30-1.png" target="_blank">8f1cb1a</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-04-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cpg_gc_cg-1" class="section level3">
<h3>cpg_GC_CG</h3>
<pre class="r"><code>subtype &lt;- &quot;cpg_GC_CG&quot;
p &lt;- control_mult_int_plot(&quot;ALL&quot;, subtype, type = &quot;re&quot;)
print(p)</code></pre>
<p><img src="figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-31-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-31-1">
Past versions of unnamed-chunk-31-1.png
</button>
</p>
<div id="fig-unnamed-chunk-31-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/fb68fdf5107911fddbd86e60528a5c9612f3488b/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-31-1.png" target="_blank">fb68fdf</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-05-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/d206789c1741f96d849c980167670cda330a86fa/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-31-1.png" target="_blank">d206789</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-04-29
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/c33ff652181aef45b799411a45ed8917457b3dbf/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-31-1.png" target="_blank">c33ff65</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-04-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="loglinear-models" class="section level2">
<h2>Loglinear Models</h2>
<p>The above analyses told us that, in general, the joint distribution
of the cell counts in the 4x4 contingency table was not the product of
the row and column marginals. This is unsurprising since we see this
exact same pattern in both the genome-wide counts and the control
counts. That is, it is already well-known that genome-wide occurrence of
dinucleotides is not predicted by the product of the frequencies of the
individual nucleotides, so it’s no surprise we see an interaction in our
singletons. But is the interaction we see in the singletons the same
that we observe in the controls? That question can be addressed using
the log-linear modeling framework employed by Zhu et al (2017). Here,
we’ll fit a model of the counts for both the singletons and the
controls, allowing for each to have their own marginal distributions at
each position, but the interaction between the two positions is shared.
We then use the deviance statistic to compare the fit of this model to
the fully saturated model which allows for different interaction terms
between the singletons and the controls.</p>
<pre class="r"><code>deviance_pair &lt;- function(population, subtype, p1, p2){
  
  data_dir &lt;- &quot;/net/snowwhite/home/beckandy/research/1000G_LSCI/output/all_count_2_pos/&quot;
  f_name &lt;- paste0(data_dir, population, &quot;/&quot;, subtype, &quot;_p&quot;, p1, &quot;_q&quot;, p2, &quot;.csv&quot;)
  df &lt;- read_csv(f_name, col_types = cols()) %&gt;%
    filter(singletons &gt; 0) %&gt;%
    select(p1, p2, singletons, controls) %&gt;%
    gather(status, n, singletons:controls)
  mod_obj &lt;- glm(n ~ (p1 + p2 + status)^2, data = df, family = poisson())
  
  return(mod_obj %&gt;% deviance)
}

deviance_pair_re &lt;- function(population, subtype, p1, p2){
  
  data_dir &lt;- &quot;/net/snowwhite/home/beckandy/research/1000G_LSCI/output/all_count_2_pos/&quot;
  f_name &lt;- paste0(data_dir, population, &quot;/&quot;, subtype, &quot;_p&quot;, p1, &quot;_q&quot;, p2, &quot;.csv&quot;)
  df &lt;- read_csv(f_name, col_types = cols()) %&gt;%
    filter(singletons &gt; 0) %&gt;%
    select(p1, p2, singletons, controls) %&gt;%
    gather(status, n, singletons:controls)
  mod_obj &lt;- glm(n ~ (p1 + p2 + status)^2, data = df, family = poisson())
  df$res &lt;- residuals(mod_obj) ^ 2
  n_singletons &lt;- df %&gt;% filter(status==&quot;singletons&quot;) %&gt;% pull(n) %&gt;% sum()
  n_controls &lt;- df %&gt;% filter(status==&quot;controls&quot;) %&gt;% pull(n) %&gt;% sum()
  df &lt;- df %&gt;%
    rowwise() %&gt;%
    mutate(re.res = ifelse(status == &quot;singletons&quot;, (res / (2*n_singletons)), (res / (2*n_controls)) ))
  re &lt;- sum(df$re.res)
  return(re)
}

deviance_plot &lt;- function(population, subtype, r_start = 1){
  p &lt;- heatmap_plot(population, subtype, deviance_pair,
                    &quot;Loglinear Model Deviance: &quot;, &quot;Deviance&quot;, r_start)
  return(p)
}

deviance_re_plot &lt;- function(population, subtype, r_start = 1){
  p &lt;- heatmap_plot(population, subtype, deviance_pair_re,
                    &quot;Loglinear Model Relative Entropy: &quot;, &quot;Relative Entropy&quot;, r_start)
  return(p)
}</code></pre>
<div id="at_cg-3" class="section level3">
<h3>AT_CG</h3>
<p><img src="figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-33-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-33-1">
Past versions of unnamed-chunk-33-1.png
</button>
</p>
<div id="fig-unnamed-chunk-33-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/fb68fdf5107911fddbd86e60528a5c9612f3488b/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-33-1.png" target="_blank">fb68fdf</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-05-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/d206789c1741f96d849c980167670cda330a86fa/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-33-1.png" target="_blank">d206789</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-04-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-33-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-33-2">
Past versions of unnamed-chunk-33-2.png
</button>
</p>
<div id="fig-unnamed-chunk-33-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/fb68fdf5107911fddbd86e60528a5c9612f3488b/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-33-2.png" target="_blank">fb68fdf</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-05-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/d206789c1741f96d849c980167670cda330a86fa/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-33-2.png" target="_blank">d206789</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-04-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="at_gc-3" class="section level3">
<h3>AT_GC</h3>
<p><img src="figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-34-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-34-1">
Past versions of unnamed-chunk-34-1.png
</button>
</p>
<div id="fig-unnamed-chunk-34-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/fb68fdf5107911fddbd86e60528a5c9612f3488b/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-34-1.png" target="_blank">fb68fdf</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-05-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/d206789c1741f96d849c980167670cda330a86fa/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-34-1.png" target="_blank">d206789</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-04-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-34-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-34-2">
Past versions of unnamed-chunk-34-2.png
</button>
</p>
<div id="fig-unnamed-chunk-34-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/fb68fdf5107911fddbd86e60528a5c9612f3488b/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-34-2.png" target="_blank">fb68fdf</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-05-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/d206789c1741f96d849c980167670cda330a86fa/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-34-2.png" target="_blank">d206789</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-04-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="at_ta-3" class="section level3">
<h3>AT_TA</h3>
<p><img src="figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-35-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-35-1">
Past versions of unnamed-chunk-35-1.png
</button>
</p>
<div id="fig-unnamed-chunk-35-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/fb68fdf5107911fddbd86e60528a5c9612f3488b/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-35-1.png" target="_blank">fb68fdf</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-05-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/d206789c1741f96d849c980167670cda330a86fa/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-35-1.png" target="_blank">d206789</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-04-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-35-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-35-2">
Past versions of unnamed-chunk-35-2.png
</button>
</p>
<div id="fig-unnamed-chunk-35-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/fb68fdf5107911fddbd86e60528a5c9612f3488b/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-35-2.png" target="_blank">fb68fdf</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-05-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/d206789c1741f96d849c980167670cda330a86fa/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-35-2.png" target="_blank">d206789</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-04-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="gc_at-3" class="section level3">
<h3>GC_AT</h3>
<p><img src="figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-36-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-36-1">
Past versions of unnamed-chunk-36-1.png
</button>
</p>
<div id="fig-unnamed-chunk-36-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/fb68fdf5107911fddbd86e60528a5c9612f3488b/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-36-1.png" target="_blank">fb68fdf</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-05-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/d206789c1741f96d849c980167670cda330a86fa/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-36-1.png" target="_blank">d206789</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-04-29
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/c33ff652181aef45b799411a45ed8917457b3dbf/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-36-1.png" target="_blank">c33ff65</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-04-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/8f1cb1a25f560e383d98dc8ab9180c403f2a90a5/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-36-1.png" target="_blank">8f1cb1a</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-04-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-36-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-36-2">
Past versions of unnamed-chunk-36-2.png
</button>
</p>
<div id="fig-unnamed-chunk-36-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/fb68fdf5107911fddbd86e60528a5c9612f3488b/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-36-2.png" target="_blank">fb68fdf</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-05-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/d206789c1741f96d849c980167670cda330a86fa/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-36-2.png" target="_blank">d206789</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-04-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cpg_gc_at-3" class="section level3">
<h3>cpg_GC_AT</h3>
<p><img src="figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-37-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-37-1">
Past versions of unnamed-chunk-37-1.png
</button>
</p>
<div id="fig-unnamed-chunk-37-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/fb68fdf5107911fddbd86e60528a5c9612f3488b/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-37-1.png" target="_blank">fb68fdf</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-05-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/d206789c1741f96d849c980167670cda330a86fa/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-37-1.png" target="_blank">d206789</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-04-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-37-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-37-2">
Past versions of unnamed-chunk-37-2.png
</button>
</p>
<div id="fig-unnamed-chunk-37-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/fb68fdf5107911fddbd86e60528a5c9612f3488b/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-37-2.png" target="_blank">fb68fdf</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-05-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/d206789c1741f96d849c980167670cda330a86fa/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-37-2.png" target="_blank">d206789</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-04-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="gc_ta-3" class="section level3">
<h3>GC_TA</h3>
<p><img src="figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-38-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-38-1">
Past versions of unnamed-chunk-38-1.png
</button>
</p>
<div id="fig-unnamed-chunk-38-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/fb68fdf5107911fddbd86e60528a5c9612f3488b/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-38-1.png" target="_blank">fb68fdf</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-05-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/d206789c1741f96d849c980167670cda330a86fa/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-38-1.png" target="_blank">d206789</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-04-29
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/c33ff652181aef45b799411a45ed8917457b3dbf/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-38-1.png" target="_blank">c33ff65</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-04-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-38-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-38-2">
Past versions of unnamed-chunk-38-2.png
</button>
</p>
<div id="fig-unnamed-chunk-38-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/fb68fdf5107911fddbd86e60528a5c9612f3488b/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-38-2.png" target="_blank">fb68fdf</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-05-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/d206789c1741f96d849c980167670cda330a86fa/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-38-2.png" target="_blank">d206789</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-04-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cpg_gc_ta-3" class="section level3">
<h3>cpg_GC_TA</h3>
<p><img src="figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-39-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-39-1">
Past versions of unnamed-chunk-39-1.png
</button>
</p>
<div id="fig-unnamed-chunk-39-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/fb68fdf5107911fddbd86e60528a5c9612f3488b/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-39-1.png" target="_blank">fb68fdf</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-05-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/d206789c1741f96d849c980167670cda330a86fa/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-39-1.png" target="_blank">d206789</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-04-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-39-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-39-2">
Past versions of unnamed-chunk-39-2.png
</button>
</p>
<div id="fig-unnamed-chunk-39-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/fb68fdf5107911fddbd86e60528a5c9612f3488b/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-39-2.png" target="_blank">fb68fdf</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-05-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/d206789c1741f96d849c980167670cda330a86fa/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-39-2.png" target="_blank">d206789</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-04-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="gc_cg-2" class="section level3">
<h3>GC_CG</h3>
<p><img src="figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-40-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-40-1">
Past versions of unnamed-chunk-40-1.png
</button>
</p>
<div id="fig-unnamed-chunk-40-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/fb68fdf5107911fddbd86e60528a5c9612f3488b/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-40-1.png" target="_blank">fb68fdf</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-05-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/d206789c1741f96d849c980167670cda330a86fa/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-40-1.png" target="_blank">d206789</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-04-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-40-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-40-2">
Past versions of unnamed-chunk-40-2.png
</button>
</p>
<div id="fig-unnamed-chunk-40-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/fb68fdf5107911fddbd86e60528a5c9612f3488b/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-40-2.png" target="_blank">fb68fdf</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-05-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/d206789c1741f96d849c980167670cda330a86fa/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-40-2.png" target="_blank">d206789</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-04-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cpg_gc_cg-2" class="section level3">
<h3>cpg_GC_CG</h3>
<p><img src="figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-41-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-41-1">
Past versions of unnamed-chunk-41-1.png
</button>
</p>
<div id="fig-unnamed-chunk-41-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/fb68fdf5107911fddbd86e60528a5c9612f3488b/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-41-1.png" target="_blank">fb68fdf</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-05-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-41-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-41-2">
Past versions of unnamed-chunk-41-2.png
</button>
</p>
<div id="fig-unnamed-chunk-41-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/fb68fdf5107911fddbd86e60528a5c9612f3488b/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-41-2.png" target="_blank">fb68fdf</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-05-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="di-nucleotide-residuals" class="section level2">
<h2>Di-nucleotide Residuals</h2>
<p>Similar to what we investigated in the single model case, here we
look into each pair of position’s deviance/RE result and break it down
into the contributions from each pair of nucleotides. Here we have a
challenge that we did not have in the single position case: for each
pair of nucleotides we have two residuals: one for the fitted singleton
value, and the other for the fitted control count. While they are both
used to calculate the overall statistic for the position, for this first
approach I am only going to look at the dinucleotide residuals for the
singleton fitted values. This is because I find this case to be more
straightforward in regards to assigning a sign and interpreting (i.e.,
do we observe more XY in the singletons than expected? etc.)</p>
<p>First let’s get the single-position level residuals for a given sub
type and population:</p>
<pre class="r"><code>get_residuals_re &lt;- function(subtype, population, p1, p2){
  data_dir &lt;- &quot;/net/snowwhite/home/beckandy/research/1000G_LSCI/output/all_count_2_pos/&quot;
  f_name &lt;- paste0(data_dir, population, &quot;/&quot;, subtype, &quot;_p&quot;, p1, &quot;_q&quot;, p2, &quot;.csv&quot;)
  df &lt;- read_csv(f_name, col_types = cols()) %&gt;%
    filter(singletons &gt; 0) %&gt;%
    select(p1, p2, singletons, controls) %&gt;%
    gather(status, n, singletons:controls)
  mod_obj &lt;- glm(n ~ (p1 + p2 + status)^2, data = df, family = poisson())
  df$res &lt;- residuals(mod_obj) ^ 2
  df$fit &lt;- predict(mod_obj, type = &quot;response&quot;)
  n_singletons &lt;- df %&gt;% filter(status==&quot;singletons&quot;) %&gt;% pull(n) %&gt;% sum()
  n_controls &lt;- df %&gt;% filter(status==&quot;controls&quot;) %&gt;% pull(n) %&gt;% sum()
  df &lt;- df %&gt;%
    rowwise() %&gt;%
    mutate(re.res = ifelse(status == &quot;singletons&quot;, (res / (2*n_singletons)), (res / (2*n_controls)) ))
  df &lt;- df %&gt;%
    filter(status == &quot;singletons&quot;) %&gt;%
    mutate(dir = sign(n - fit)) %&gt;%
    mutate(s.res = re.res * dir) %&gt;%
    ungroup() %&gt;%
    mutate(prop.res = sqrt(res) / sum(sqrt(res))) %&gt;%
    arrange(desc(res))
  return(df)
}</code></pre>
<p>Now, as an example let’s look at the residuals for the A&gt;G sub
type at the -2, -1 positions:</p>
<pre class="r"><code>get_residuals_re(&quot;AT_GC&quot;, &quot;ALL&quot;, -2, -1) %&gt;%
  knitr::kable()</code></pre>
<table>
<colgroup>
<col width="3%" />
<col width="3%" />
<col width="13%" />
<col width="9%" />
<col width="14%" />
<col width="12%" />
<col width="12%" />
<col width="4%" />
<col width="13%" />
<col width="12%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">p1</th>
<th align="left">p2</th>
<th align="left">status</th>
<th align="right">n</th>
<th align="right">res</th>
<th align="right">fit</th>
<th align="right">re.res</th>
<th align="right">dir</th>
<th align="right">s.res</th>
<th align="right">prop.res</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">C</td>
<td align="left">A</td>
<td align="left">singletons</td>
<td align="right">1504194</td>
<td align="right">67046.22613</td>
<td align="right">1208568.6</td>
<td align="right">0.0024488</td>
<td align="right">1</td>
<td align="right">0.0024488</td>
<td align="right">0.1732595</td>
</tr>
<tr class="even">
<td align="left">C</td>
<td align="left">C</td>
<td align="left">singletons</td>
<td align="right">1069119</td>
<td align="right">65582.98838</td>
<td align="right">1356210.8</td>
<td align="right">0.0023954</td>
<td align="right">-1</td>
<td align="right">-0.0023954</td>
<td align="right">0.1713584</td>
</tr>
<tr class="odd">
<td align="left">A</td>
<td align="left">C</td>
<td align="left">singletons</td>
<td align="right">1218930</td>
<td align="right">26111.23668</td>
<td align="right">1049122.1</td>
<td align="right">0.0009537</td>
<td align="right">1</td>
<td align="right">0.0009537</td>
<td align="right">0.1081244</td>
</tr>
<tr class="even">
<td align="left">A</td>
<td align="left">A</td>
<td align="left">singletons</td>
<td align="right">1246030</td>
<td align="right">24974.74913</td>
<td align="right">1430857.9</td>
<td align="right">0.0009122</td>
<td align="right">-1</td>
<td align="right">-0.0009122</td>
<td align="right">0.1057452</td>
</tr>
<tr class="odd">
<td align="left">C</td>
<td align="left">G</td>
<td align="left">singletons</td>
<td align="right">64649</td>
<td align="right">21941.86679</td>
<td align="right">109954.8</td>
<td align="right">0.0008014</td>
<td align="right">-1</td>
<td align="right">-0.0008014</td>
<td align="right">0.0991167</td>
</tr>
<tr class="even">
<td align="left">T</td>
<td align="left">C</td>
<td align="left">singletons</td>
<td align="right">925793</td>
<td align="right">11624.76977</td>
<td align="right">825890.5</td>
<td align="right">0.0004246</td>
<td align="right">1</td>
<td align="right">0.0004246</td>
<td align="right">0.0721443</td>
</tr>
<tr class="odd">
<td align="left">G</td>
<td align="left">A</td>
<td align="left">singletons</td>
<td align="right">633750</td>
<td align="right">7982.57238</td>
<td align="right">707561.7</td>
<td align="right">0.0002916</td>
<td align="right">-1</td>
<td align="right">-0.0002916</td>
<td align="right">0.0597835</td>
</tr>
<tr class="even">
<td align="left">T</td>
<td align="left">T</td>
<td align="left">singletons</td>
<td align="right">869379</td>
<td align="right">4972.60515</td>
<td align="right">936797.0</td>
<td align="right">0.0001816</td>
<td align="right">-1</td>
<td align="right">-0.0001816</td>
<td align="right">0.0471848</td>
</tr>
<tr class="odd">
<td align="left">G</td>
<td align="left">T</td>
<td align="left">singletons</td>
<td align="right">669774</td>
<td align="right">4604.49110</td>
<td align="right">615764.6</td>
<td align="right">0.0001682</td>
<td align="right">1</td>
<td align="right">0.0001682</td>
<td align="right">0.0454047</td>
</tr>
<tr class="even">
<td align="left">T</td>
<td align="left">A</td>
<td align="left">singletons</td>
<td align="right">611645</td>
<td align="right">2150.24768</td>
<td align="right">648630.8</td>
<td align="right">0.0000785</td>
<td align="right">-1</td>
<td align="right">-0.0000785</td>
<td align="right">0.0310280</td>
</tr>
<tr class="odd">
<td align="left">A</td>
<td align="left">G</td>
<td align="left">singletons</td>
<td align="right">819729</td>
<td align="right">1855.45979</td>
<td align="right">781345.3</td>
<td align="right">0.0000678</td>
<td align="right">1</td>
<td align="right">0.0000678</td>
<td align="right">0.0288228</td>
</tr>
<tr class="even">
<td align="left">C</td>
<td align="left">T</td>
<td align="left">singletons</td>
<td align="right">1163980</td>
<td align="right">1186.76523</td>
<td align="right">1127207.7</td>
<td align="right">0.0000433</td>
<td align="right">1</td>
<td align="right">0.0000433</td>
<td align="right">0.0230511</td>
</tr>
<tr class="odd">
<td align="left">A</td>
<td align="left">T</td>
<td align="left">singletons</td>
<td align="right">1110832</td>
<td align="right">484.61429</td>
<td align="right">1134195.7</td>
<td align="right">0.0000177</td>
<td align="right">-1</td>
<td align="right">-0.0000177</td>
<td align="right">0.0147302</td>
</tr>
<tr class="even">
<td align="left">G</td>
<td align="left">C</td>
<td align="left">singletons</td>
<td align="right">705354</td>
<td align="right">435.48575</td>
<td align="right">687972.5</td>
<td align="right">0.0000159</td>
<td align="right">1</td>
<td align="right">0.0000159</td>
<td align="right">0.0139636</td>
</tr>
<tr class="odd">
<td align="left">T</td>
<td align="left">G</td>
<td align="left">singletons</td>
<td align="right">567955</td>
<td align="right">35.86432</td>
<td align="right">563453.7</td>
<td align="right">0.0000013</td>
<td align="right">1</td>
<td align="right">0.0000013</td>
<td align="right">0.0040072</td>
</tr>
<tr class="even">
<td align="left">G</td>
<td align="left">G</td>
<td align="left">singletons</td>
<td align="right">508288</td>
<td align="right">11.56687</td>
<td align="right">505867.1</td>
<td align="right">0.0000004</td>
<td align="right">1</td>
<td align="right">0.0000004</td>
<td align="right">0.0022757</td>
</tr>
</tbody>
</table>
<p>Here we see that among the two position influences the biggest
difference between the singleton and the control distributions was for
the CA dinucleotide; this is the same as what Zhu et al (2017) observed
at this position for this sub type. Let’s take a look at what happens if
we summarize each pair of positions by the sum of the two re/deviance
residuals instead of just ignoring the control residual:</p>
<pre class="r"><code>get_residuals_re2 &lt;- function(subtype, population, p1, p2){
  data_dir &lt;- &quot;/net/snowwhite/home/beckandy/research/1000G_LSCI/output/all_count_2_pos/&quot;
  f_name &lt;- paste0(data_dir, population, &quot;/&quot;, subtype, &quot;_p&quot;, p1, &quot;_q&quot;, p2, &quot;.csv&quot;)
  df &lt;- read_csv(f_name, col_types = cols()) %&gt;%
    filter(singletons &gt; 0) %&gt;%
    select(p1, p2, singletons, controls) %&gt;%
    gather(status, n, singletons:controls)
  mod_obj &lt;- glm(n ~ (p1 + p2 + status)^2, data = df, family = poisson())
  df$res &lt;- residuals(mod_obj) ^ 2
  df$fit &lt;- predict(mod_obj, type = &quot;response&quot;)
  n_singletons &lt;- df %&gt;% filter(status==&quot;singletons&quot;) %&gt;% pull(n) %&gt;% sum()
  n_controls &lt;- df %&gt;% filter(status==&quot;controls&quot;) %&gt;% pull(n) %&gt;% sum()
  df &lt;- df %&gt;%
    rowwise() %&gt;%
    mutate(re.res = ifelse(status == &quot;singletons&quot;, (res / (2*n_singletons)), (res / (2*n_controls)) ))
  df &lt;- df %&gt;%
    select(p1, p2, re.res) %&gt;%
    group_by(p1, p2) %&gt;%
    summarize(re.res = sum(re.res)) %&gt;%
    ungroup() %&gt;%
    mutate(prop.re = re.res / sum(re.res)) %&gt;%
    arrange(desc(re.res))
  return(df)
}</code></pre>
<p>Here again we see that the top contribution for any dinucleotide pair
is CT at the -2, -1 positions when we account for both the singleton and
the control residuals:</p>
<pre class="r"><code>get_residuals_re2(&quot;AT_GC&quot;, &quot;ALL&quot;, -2, -1) %&gt;%
  knitr::kable()</code></pre>
<pre><code>`summarise()` has grouped output by &#39;p1&#39;. You can override using the `.groups`
argument.</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">p1</th>
<th align="left">p2</th>
<th align="right">re.res</th>
<th align="right">prop.re</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">C</td>
<td align="left">A</td>
<td align="right">0.0025912</td>
<td align="right">0.2804169</td>
</tr>
<tr class="even">
<td align="left">C</td>
<td align="left">C</td>
<td align="right">0.0025376</td>
<td align="right">0.2746151</td>
</tr>
<tr class="odd">
<td align="left">A</td>
<td align="left">C</td>
<td align="right">0.0009988</td>
<td align="right">0.1080905</td>
</tr>
<tr class="even">
<td align="left">A</td>
<td align="left">A</td>
<td align="right">0.0009431</td>
<td align="right">0.1020622</td>
</tr>
<tr class="odd">
<td align="left">C</td>
<td align="left">G</td>
<td align="right">0.0008316</td>
<td align="right">0.0899985</td>
</tr>
<tr class="even">
<td align="left">T</td>
<td align="left">C</td>
<td align="right">0.0004397</td>
<td align="right">0.0475814</td>
</tr>
<tr class="odd">
<td align="left">G</td>
<td align="left">A</td>
<td align="right">0.0003006</td>
<td align="right">0.0325263</td>
</tr>
<tr class="even">
<td align="left">T</td>
<td align="left">T</td>
<td align="right">0.0001882</td>
<td align="right">0.0203663</td>
</tr>
<tr class="odd">
<td align="left">G</td>
<td align="left">T</td>
<td align="right">0.0001758</td>
<td align="right">0.0190270</td>
</tr>
<tr class="even">
<td align="left">T</td>
<td align="left">A</td>
<td align="right">0.0000806</td>
<td align="right">0.0087263</td>
</tr>
<tr class="odd">
<td align="left">A</td>
<td align="left">G</td>
<td align="right">0.0000699</td>
<td align="right">0.0075653</td>
</tr>
<tr class="even">
<td align="left">C</td>
<td align="left">T</td>
<td align="right">0.0000465</td>
<td align="right">0.0050332</td>
</tr>
<tr class="odd">
<td align="left">A</td>
<td align="left">T</td>
<td align="right">0.0000186</td>
<td align="right">0.0020080</td>
</tr>
<tr class="even">
<td align="left">G</td>
<td align="left">C</td>
<td align="right">0.0000165</td>
<td align="right">0.0017909</td>
</tr>
<tr class="odd">
<td align="left">T</td>
<td align="left">G</td>
<td align="right">0.0000013</td>
<td align="right">0.0001451</td>
</tr>
<tr class="even">
<td align="left">G</td>
<td align="left">G</td>
<td align="right">0.0000004</td>
<td align="right">0.0000470</td>
</tr>
</tbody>
</table>
<p>Perhaps we’d like to visualize these results in a heatmap?</p>
<pre class="r"><code>heatmap_re_res &lt;- function(subtype, pop, p1, p2, singletons_only = TRUE){
  if(singletons_only){
    df &lt;- get_residuals_re(subtype, pop, p1, p2)
  } else {
    df &lt;- get_residuals_re2(subtype, pop, p1, p2)
  }
  p &lt;- df %&gt;%
    ggplot(aes(x = p2, y = p1, fill = re.res)) +
    geom_tile() +
    ggtitle(paste0(&quot;Interaction RE: &quot;, subtype),
            paste0(&quot;Population: &quot;, pop))+
    xlab(paste0(&quot;Relative Position: &quot;, p2)) +
    ylab(paste0(&quot;Relative Position: &quot;, p1)) +
    labs(fill = &quot;RE Residual&quot;) +
    scale_fill_distiller(palette = &quot;Reds&quot;, direction = 1)
  return(p)
}

heatmap_signed_re_res &lt;- function(subtype, pop, p1, p2){
  df &lt;- get_residuals_re(subtype, pop, p1, p2)
  p &lt;- df %&gt;%
    ggplot(aes(x = p2, y = p1, fill = s.res)) +
    geom_tile() +
    ggtitle(paste0(&quot;Interaction RE: &quot;, subtype),
            paste0(&quot;Population: &quot;, pop))+
    xlab(paste0(&quot;Relative Position: &quot;, p2)) +
    ylab(paste0(&quot;Relative Position: &quot;, p1)) +
    labs(fill = &quot;RE Residual&quot;) +
    scale_fill_distiller(palette = &quot;RdBu&quot;, direction = -1)
  return(p)

}</code></pre>
<p>And now we plot the signed re residuals for the A&gt;G sub type:</p>
<pre class="r"><code>heatmap_signed_re_res(&quot;AT_GC&quot;, &quot;ALL&quot;, -2, -1)</code></pre>
<p><img src="figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-47-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-47-1">
Past versions of unnamed-chunk-47-1.png
</button>
</p>
<div id="fig-unnamed-chunk-47-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/fb68fdf5107911fddbd86e60528a5c9612f3488b/docs/figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-47-1.png" target="_blank">fb68fdf</a>
</td>
<td>
Andy Beck
</td>
<td>
2022-05-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="alternative-di-nucleotide-residuals" class="section level2">
<h2>Alternative Di-nucleotide Residuals</h2>
<div id="re-contribution-control-intercept-estimation"
class="section level3">
<h3>RE Contribution (Control Intercept Estimation)</h3>
<p>In the analysis above where we used the control samples to estimate
the interaction and then adjusted marginal products using this
interaction, we computed the relative entropy between our observed
singleton distribution and the expectation under marginal independence
adjusted by control interaction as our measure of influence. This
statistic is the sum of expected logarithmic differences for each cell
in the 4x4 table, and we will look at these individual values to assess
at each pair of positions which di-nucleotides were either over or
under-represented. We can also look at values for individual nucleotides
at one position summed over the nucleotides at the other position (not
sure how this would be interpreted, to be honest).</p>
<pre class="r"><code>control_int_mult_res &lt;- function(population, subtype, p1, p2) {
  data_dir &lt;- &quot;/net/snowwhite/home/beckandy/research/1000G_LSCI/output/all_count_2_pos/&quot;
  f_name &lt;- paste0(data_dir, population, &quot;/&quot;, subtype, &quot;_p&quot;, p1, &quot;_q&quot;, p2, &quot;.csv&quot;)
  df &lt;- read_csv(f_name, col_types = cols()) %&gt;%
    filter(singletons &gt; 0)
  control_tab &lt;- df %&gt;% 
    select(p1, p2, controls) %&gt;% 
    pivot_wider(names_from = p2, values_from = controls) %&gt;%
    remove_rownames() %&gt;%
    column_to_rownames(var = &#39;p1&#39;)
  singleton_tab &lt;- df %&gt;% 
    select(p1, p2, singletons) %&gt;% 
    pivot_wider(names_from = p2, values_from = singletons) %&gt;%
    remove_rownames() %&gt;%
    column_to_rownames(var = &#39;p1&#39;)

  exp_s &lt;- get_mar(singleton_tab) * get_int(control_tab)
  exp_p_s &lt;- exp_s / sum(exp_s) # normalize
  exp_s &lt;- exp_p_s * sum(singleton_tab)
  #chi_sq &lt;- (exp_s - singleton_tab)^2 / exp_s
  
  p_s &lt;- singleton_tab / sum(singleton_tab)
  re &lt;- p_s * log(p_s / exp_p_s)
  final &lt;- rownames_to_column(re, &quot;p1&quot;) %&gt;% 
    pivot_longer(-p1, names_to = &quot;p2&quot;, values_to = &quot;statistic&quot;)
  
  return(final)
}

plot_control_mult_res &lt;- function(population, subtype, p1, p2){
  df &lt;- control_int_mult_res(population, subtype, p1, p2)
  p &lt;- df %&gt;%
    ggplot(aes(x = p2, y = p1, fill = statistic)) +
    geom_tile() +
    scale_fill_distiller(palette = &quot;RdBu&quot;) +
    ggtitle(&quot;Singleton di-nucleotide Enrichment/Depletion&quot;) +
    xlab(&quot;Relative Position 2&quot;) + 
    ylab(&quot;Relative Position 1&quot;) +
    labs(fill=&quot;RE&quot;)
  return(p)
}</code></pre>
<p>So, again returning to the example of -2, -1 in AT_GC:</p>
<pre class="r"><code>plot_control_mult_res(&quot;ALL&quot;, &quot;AT_GC&quot;, -2, -1)</code></pre>
<p><img src="figure/1000G_two_positions_ALL.Rmd/unnamed-chunk-49-1.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 18.04.6 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/openblas/libblas.so.3
LAPACK: /usr/lib/x86_64-linux-gnu/libopenblasp-r0.2.20.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] tikzDevice_0.12.3.1 forcats_0.5.1       stringr_1.4.0      
 [4] dplyr_1.0.8         purrr_0.3.4         readr_2.1.2        
 [7] tidyr_1.2.0         tibble_3.1.6        ggplot2_3.3.5      
[10] tidyverse_1.3.1     workflowr_1.7.0    

loaded via a namespace (and not attached):
 [1] httr_1.4.2         sass_0.4.0         bit64_4.0.5        vroom_1.5.7       
 [5] jsonlite_1.7.3     modelr_0.1.8       bslib_0.3.1        assertthat_0.2.1  
 [9] getPass_0.2-2      highr_0.9          cellranger_1.1.0   yaml_2.2.2        
[13] pillar_1.7.0       backports_1.4.1    glue_1.6.1         digest_0.6.29     
[17] RColorBrewer_1.1-2 promises_1.2.0.1   rvest_1.0.2        colorspace_2.0-2  
[21] htmltools_0.5.2    httpuv_1.6.5       pkgconfig_2.0.3    broom_0.8.0       
[25] haven_2.4.3        scales_1.1.1       processx_3.5.2     whisker_0.4       
[29] later_1.3.0        tzdb_0.2.0         git2r_0.29.0       generics_0.1.2    
[33] farver_2.1.0       ellipsis_0.3.2     withr_2.4.3        cli_3.1.1         
[37] magrittr_2.0.2     crayon_1.4.2       readxl_1.3.1       evaluate_0.14     
[41] ps_1.6.0           fs_1.5.2           fansi_1.0.2        xml2_1.3.3        
[45] tools_4.2.0        hms_1.1.1          lifecycle_1.0.1    munsell_0.5.0     
[49] reprex_2.0.1       callr_3.7.0        compiler_4.2.0     jquerylib_0.1.4   
[53] rlang_1.0.1        grid_4.2.0         rstudioapi_0.13    filehash_2.4-2    
[57] labeling_0.4.2     rmarkdown_2.11     gtable_0.3.0       DBI_1.1.2         
[61] R6_2.5.1           lubridate_1.8.0    knitr_1.37         fastmap_1.1.0     
[65] bit_4.0.4          utf8_1.2.2         rprojroot_2.0.2    stringi_1.7.6     
[69] parallel_4.2.0     Rcpp_1.0.8         vctrs_0.3.8        dbplyr_2.1.1      
[73] tidyselect_1.1.1   xfun_0.29         </code></pre>
</div>
</div>
</div>
</div>

<div id="disqus_thread"></div>
<script>
(function() { // DON'T EDIT BELOW THIS LINE
var d = document, s = d.createElement('script');
s.src = 'https://lsci-doc.disqus.com/embed.js';
s.setAttribute('data-timestamp', +new Date());
(d.head || d.body).appendChild(s);
})();
</script>
<noscript>Please enable JavaScript to view the
<a href="https://disqus.com/?ref_noscript">
  comments powered by Disqus.</a></noscript>

<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
