<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Andy Beck" />

<meta name="date" content="2023-05-23" />

<title>NYGC_single_position_ALL</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Local Sequence Context Influence</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/theandyb/LSCI">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">NYGC_single_position_ALL</h1>
<h4 class="author">Andy Beck</h4>
<h4 class="date">2023-05-23</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-05-31
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>LSCI/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20211101code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20211101)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20211101code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20211101)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomtheandybLSCItreea1ed7221e986cc885c018ca1b3be8b48edd87ca3targetblanka1ed722a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/theandyb/LSCI/tree/a1ed7221e986cc885c018ca1b3be8b48edd87ca3" target="_blank">a1ed722</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomtheandybLSCItreea1ed7221e986cc885c018ca1b3be8b48edd87ca3targetblanka1ed722a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/theandyb/LSCI/tree/a1ed7221e986cc885c018ca1b3be8b48edd87ca3" target="_blank">a1ed722</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/disqus.html
    Ignored:    code/random/
    Ignored:    output/figs/

Unstaged changes:
    Modified:   analysis/NYGC_single_position_comp.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/NYGC_single_position_ALL.Rmd</code>) and HTML
(<code>docs/NYGC_single_position_ALL.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/theandyb/LSCI/blob/a1ed7221e986cc885c018ca1b3be8b48edd87ca3/analysis/NYGC_single_position_ALL.Rmd" target="_blank">a1ed722</a>
</td>
<td>
Andy Beck
</td>
<td>
2023-05-31
</td>
<td>
wflow_publish("analysis/NYGC_single_position_ALL.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/theandyb/LSCI/04837fdc1d6e457e8e4b0832bdc731e79954c9f4/docs/NYGC_single_position_ALL.html" target="_blank">04837fd</a>
</td>
<td>
Andy Beck
</td>
<td>
2023-05-31
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/theandyb/LSCI/blob/9f03e7ef854ab058e22c9ddd65fd6a54d311d337/analysis/NYGC_single_position_ALL.Rmd" target="_blank">9f03e7e</a>
</td>
<td>
Andy Beck
</td>
<td>
2023-05-31
</td>
<td>
wflow_publish("analysis/NYGC_single_position_ALL.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/theandyb/LSCI/457d3ee938b4ef56dd57aad06102d34f129b2311/docs/NYGC_single_position_ALL.html" target="_blank">457d3ee</a>
</td>
<td>
Andy Beck
</td>
<td>
2023-05-31
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/theandyb/LSCI/blob/d0036044fc067db96addd17aa6700118978b955f/analysis/NYGC_single_position_ALL.Rmd" target="_blank">d003604</a>
</td>
<td>
Andy Beck
</td>
<td>
2023-05-31
</td>
<td>
wflow_publish("analysis/NYGC_single_position_ALL.Rmd")
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/theandyb/LSCI/blob/8bec6163d036587514bc90ec53210d7543cd3a7d/analysis/NYGC_single_position_ALL.Rmd" target="_blank">8bec616</a>
</td>
<td>
Andy Beck
</td>
<td>
2023-05-25
</td>
<td>
new docs
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/theandyb/LSCI/b3ea038581eafb29e23eb842e7b702b724257791/docs/NYGC_single_position_ALL.html" target="_blank">b3ea038</a>
</td>
<td>
Andy Beck
</td>
<td>
2023-05-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/theandyb/LSCI/blob/372210004bb0d535238ac0201e823be6f8e5801f/analysis/NYGC_single_position_ALL.Rmd" target="_blank">3722100</a>
</td>
<td>
Andy Beck
</td>
<td>
2023-05-24
</td>
<td>
wflow_publish("analysis/NYGC_single_position_ALL.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/theandyb/LSCI/f77e94cfaf9173a4c9361439ca41b972e9b44c8e/docs/NYGC_single_position_ALL.html" target="_blank">f77e94c</a>
</td>
<td>
Andy Beck
</td>
<td>
2023-05-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/theandyb/LSCI/blob/e32b997bf2754046c4a4512bec17660afe08759f/analysis/NYGC_single_position_ALL.Rmd" target="_blank">e32b997</a>
</td>
<td>
Andy Beck
</td>
<td>
2023-05-23
</td>
<td>
wflow_publish(c("analysis/index.Rmd",
"analysis/NYGC_single_position_ALL.Rmd"))
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="g-single-position-models---all" class="section level1">
<h1>1000G Single Position Models - All</h1>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>Here in this document we present the results for the single position
models using the singletons from all 2,504 unrelated individuals in the
1000G deep sequence data from the NYGC.</p>
<div id="model-descriptions" class="section level3">
<h3>Model Descriptions</h3>
</div>
<div id="log-linear-model-approach" class="section level3">
<h3>Log Linear Model Approach</h3>
<p>Following the example of Zhu et al (2017), we also construct models
of the form</p>
<p><span class="math display">\[
\log(n_{ij}) = \lambda_0 + \sum_{k \in \{C,G,T\}} \lambda_j I(i=k) +
\lambda_{s}I(j=s)
\]</span></p>
<p>where we jointly model the counts for singletons and their matched
controls stratified by the nucleotides at a given flanking position in
the +/- 10 bp window. Note that here the model assumes a shared
nucleotide distribution between the singletons and the controls;
allowing for a difference through an interaction term would yield a
fully saturated model. Zhu et al (2017) use the deviance statistic to
identify putative mutation motifs, which compares the above reduced
model to the fully saturated model, with a larger deviance statistic
resulting from a larger difference between the observed distribution of
nucleotides flanking singleton and control sites from the fitted
distribution based on the reduced model. The deviance statistic tends to
increase due to sample size, so Zhu et al (2017) additionally divided
the deviance statistic by twice the sample size to yield the relative
entropy/KL divergence, which is not sensitive to sample size and allowed
for comparisons between mutation sub types. We choose to take a similar
approach in that we compute the KL divergence between the observed and
fitted frequencies for only the singletons.</p>
</div>
</div>
<div id="subtype-results" class="section level2">
<h2>9 Subtype Results</h2>
<p>The nine basic subtypes are AT_CG, AT_GC, AT_TA, GC_AT, GC_TA, GC_CG,
cpg_GC_AT, cpg_GC_TA, and cpg_GC_CG, where the notation XY_JK indicates
we either observed a X&gt;J or Y&gt;K substitution, with XY and JK being
complementary to each other. For each sub-type with reference alleles XY
and position i within a 21-mer window centered at the singleton, we
model the distributions of nucleotides A, C, G and T flanking the
singletons using the rates we observe either genome-wide flanking the
reference nucleotides XY at the same relative position or the rates of
nucleotides flanking control positions sampled for each singleton. We
use the chi-square goodness of fit statistic to “measure” the degree to
which this model doesn’t hold. First, let’s plot the deviance statistics
we observe at each relative position for each sub-type.</p>
<div id="code-for-loading-results" class="section level3">
<h3>Code for loading results</h3>
<pre class="r"><code>read_subtype_pop_res &lt;- function(pop, subtype, data_dir = &quot;/net/snowwhite/home/beckandy/research/1000G_NYGC_LSCI/output/single_pos/&quot;){
  fname &lt;- paste0(data_dir, pop, &quot;/&quot;, subtype, &quot;.csv&quot;)
  df &lt;- read_csv(fname, show_col_types = FALSE) %&gt;%
    rowwise() %&gt;%
    mutate(re = dev / (2 * (singletons + controls)))
  return(df)
}

read_pop_res &lt;- function(pop, data_dir = &quot;/net/snowwhite/home/beckandy/research/1000G_NYGC_LSCI/output/single_pos/&quot;){
  df &lt;- read_subtype_pop_res(pop, &quot;AT_CG&quot;, data_dir = data_dir)
  df$subtype &lt;- &quot;AT_CG&quot;
  for(st in c(&quot;AT_GC&quot;, &quot;AT_TA&quot;, 
              &quot;GC_AT&quot;, &quot;GC_TA&quot;, &quot;GC_CG&quot;,
              &quot;cpg_GC_AT&quot;, &quot;cpg_GC_TA&quot;, &quot;cpg_GC_CG&quot;) ) {
    df2 &lt;- read_subtype_pop_res(pop, st, data_dir = data_dir)
    df2$subtype &lt;- st
    df &lt;- bind_rows(df, df2)
  }
  return(df)
}

read_subtype_pop_resid &lt;- function(pop, subtype, rp, data_dir = &quot;/net/snowwhite/home/beckandy/research/1000G_NYGC_LSCI/output/single_pos/resid/&quot;){
  f_name &lt;- paste0(data_dir, pop, &quot;/&quot;, subtype, &quot;_rp_&quot;, rp, &quot;.csv&quot;)
  df &lt;- read_csv(f_name,show_col_types = FALSE)
  return(df)
}

read_subtype_pop_range_resid &lt;- function(pop, subtype, size = 10, data_dir = &quot;/net/snowwhite/home/beckandy/research/1000G_NYGC_LSCI/output/single_pos/resid/&quot;){
  if(str_starts(subtype, &quot;cpg&quot;)){
    pos.names &lt;- c(-size:-1, 2:size)
  } else{
    pos.names &lt;- c(-size:-1, 1:size)
  }
  results &lt;- lapply(pos.names, function(x) NULL) %&gt;% setNames(pos.names)
  for(i in names(results)){
    results[[i]] &lt;- read_subtype_pop_resid(pop, subtype, i, data_dir)
    results[[i]]$pos &lt;- i
  }
  df &lt;- bind_rows(results)
  return(df)
}

subtype_print_names &lt;- function(st){
  if(str_starts(st, &quot;AT&quot;)){
    return(paste0(&quot;A → &quot;, str_sub(st, 4, 4)))
  } else if(str_starts(st, &quot;GC&quot;)){
    return(paste0(&quot;C → &quot;, str_sub(st, 5, 5)))
  } else{
    return(paste0(&quot;CpG → &quot;, str_sub(st, 9, 9), &quot;pG&quot;))
  }
}</code></pre>
</div>
<div id="all-subtypes-plot-options" class="section level3">
<h3>All subtypes plot options</h3>
<pre class="r"><code>df &lt;- read_pop_res(&quot;ALL&quot;) %&gt;%
  rowwise() %&gt;%
  mutate(print_name = subtype_print_names(subtype))

df %&gt;%
  filter(abs(offset) &lt; 10) %&gt;%
  ggplot(aes(x = offset, y = re, color = print_name)) +
  geom_point() + 
  geom_line(show.legend = FALSE) +
  scale_color_manual(values = cbPalette) +
  xlab(&quot;Relative Position&quot;) +
  ylab(&quot;Relative Entropy&quot;) +
  labs(color = &quot;Subtype&quot;) +
  theme_bw(base_size = 18/.pt) +
  theme(text = element_text(family = &quot;Times New Roman&quot;)) +
  theme(
    legend.position = c(.98, .98),
    legend.justification = c(&quot;right&quot;, &quot;top&quot;),
    legend.box.just = &quot;right&quot;
    ) </code></pre>
<p><img src="figure/NYGC_single_position_ALL.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-1">
Past versions of unnamed-chunk-3-1.png
</button>
</p>
<div id="fig-unnamed-chunk-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/457d3ee938b4ef56dd57aad06102d34f129b2311/docs/figure/NYGC_single_position_ALL.Rmd/unnamed-chunk-3-1.png" target="_blank">457d3ee</a>
</td>
<td>
Andy Beck
</td>
<td>
2023-05-31
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/b3ea038581eafb29e23eb842e7b702b724257791/docs/figure/NYGC_single_position_ALL.Rmd/unnamed-chunk-3-1.png" target="_blank">b3ea038</a>
</td>
<td>
Andy Beck
</td>
<td>
2023-05-24
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/f77e94cfaf9173a4c9361439ca41b972e9b44c8e/docs/figure/NYGC_single_position_ALL.Rmd/unnamed-chunk-3-1.png" target="_blank">f77e94c</a>
</td>
<td>
Andy Beck
</td>
<td>
2023-05-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggsave(paste0(plot_dir, &quot;fig1_opt1.tiff&quot;), dpi = 300, height = 4, width = 5)</code></pre>
<p>Separate plots for A&gt;N and C&gt;N</p>
<pre class="r"><code>df %&gt;%
  filter(abs(offset) &lt; 10,
         str_starts(subtype, &quot;AT&quot;)) %&gt;%
  ggplot(aes(x = offset, y = re, color = print_name)) +
  geom_point() + 
  geom_line(show.legend = FALSE) +
  scale_color_manual(values = cbPalette) +
  xlab(&quot;Relative Position&quot;) +
  ylab(&quot;Relative Entropy&quot;) +
  labs(color = &quot;Subtype&quot;) +
  theme_classic(base_size = 18/.pt) +
  theme(text = element_text(family = &quot;Times New Roman&quot;),
        legend.position = c(.90, .85))</code></pre>
<p><img src="figure/NYGC_single_position_ALL.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-1">
Past versions of unnamed-chunk-4-1.png
</button>
</p>
<div id="fig-unnamed-chunk-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/b3ea038581eafb29e23eb842e7b702b724257791/docs/figure/NYGC_single_position_ALL.Rmd/unnamed-chunk-4-1.png" target="_blank">b3ea038</a>
</td>
<td>
Andy Beck
</td>
<td>
2023-05-24
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/f77e94cfaf9173a4c9361439ca41b972e9b44c8e/docs/figure/NYGC_single_position_ALL.Rmd/unnamed-chunk-4-1.png" target="_blank">f77e94c</a>
</td>
<td>
Andy Beck
</td>
<td>
2023-05-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggsave(paste0(plot_dir, &quot;fig1_opt2a.tiff&quot;), dpi = plot_dpi, height = 4, width = 5)</code></pre>
<pre class="r"><code>df %&gt;%
  filter(abs(offset) &lt; 10,
         !str_starts(subtype, &quot;AT&quot;)) %&gt;%
  ggplot(aes(x = offset, y = re, color = print_name)) +
  geom_point() + 
  geom_line() +
  scale_color_manual(values = cbPalette) +
  xlab(&quot;Relative Position&quot;) +
  ylab(&quot;Relative Entropy&quot;) +
  labs(color = &quot;Subtype&quot;) +
  theme_classic(base_size = 18/.pt) +
  theme(text = element_text(family = &quot;Times New Roman&quot;),
        legend.position = c(.90, .75))</code></pre>
<p><img src="figure/NYGC_single_position_ALL.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/457d3ee938b4ef56dd57aad06102d34f129b2311/docs/figure/NYGC_single_position_ALL.Rmd/unnamed-chunk-5-1.png" target="_blank">457d3ee</a>
</td>
<td>
Andy Beck
</td>
<td>
2023-05-31
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/b3ea038581eafb29e23eb842e7b702b724257791/docs/figure/NYGC_single_position_ALL.Rmd/unnamed-chunk-5-1.png" target="_blank">b3ea038</a>
</td>
<td>
Andy Beck
</td>
<td>
2023-05-24
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/f77e94cfaf9173a4c9361439ca41b972e9b44c8e/docs/figure/NYGC_single_position_ALL.Rmd/unnamed-chunk-5-1.png" target="_blank">f77e94c</a>
</td>
<td>
Andy Beck
</td>
<td>
2023-05-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggsave(paste0(plot_dir, &quot;fig1_opt2b.tiff&quot;), dpi = plot_dpi, height = 4, width = 5)</code></pre>
<p>Patchwork the plots together:</p>
<pre class="r"><code>p1 &lt;- df %&gt;%
  filter(abs(offset) &lt; 10,
         str_starts(subtype, &quot;AT&quot;)) %&gt;%
  ggplot(aes(x = offset, y = re, color = print_name)) +
  geom_point() + 
  geom_line(show.legend = FALSE) +
  scale_color_manual(values = cbPalette) +
  xlab(&quot;Relative Position&quot;) +
  ylab(&quot;Relative Entropy&quot;) +
  labs(color = &quot;Subtype&quot;) +
  theme_classic(base_size = 18/.pt) +
  theme(text = element_text(family = &quot;Times New Roman&quot;),
        legend.position = c(.90, .75))
  
p2 &lt;- df %&gt;%
  filter(abs(offset) &lt; 10,
         !str_starts(subtype, &quot;AT&quot;)) %&gt;%
  ggplot(aes(x = offset, y = re, color = print_name)) +
  geom_point() + 
  geom_line(show.legend = FALSE) +
  scale_color_manual(values = cbPalette) +
  xlab(&quot;Relative Position&quot;) +
  ylab(&quot;Relative Entropy&quot;) +
  labs(color = &quot;Subtype&quot;) +
  theme_classic(base_size = 18/.pt) +
  theme(text = element_text(family = &quot;Times New Roman&quot;),
        legend.position = c(.90, .6))

p_combined &lt;- p1 / p2

p_ranges_y &lt;- c(ggplot_build(p_combined[[1]])$layout$panel_scales_y[[1]]$range$range,
                ggplot_build(p_combined[[2]])$layout$panel_scales_y[[1]]$range$range)
p_combined &amp; 
  ylim(min(p_ranges_y), max(p_ranges_y))</code></pre>
<p><img src="figure/NYGC_single_position_ALL.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-1">
Past versions of unnamed-chunk-6-1.png
</button>
</p>
<div id="fig-unnamed-chunk-6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/457d3ee938b4ef56dd57aad06102d34f129b2311/docs/figure/NYGC_single_position_ALL.Rmd/unnamed-chunk-6-1.png" target="_blank">457d3ee</a>
</td>
<td>
Andy Beck
</td>
<td>
2023-05-31
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/b3ea038581eafb29e23eb842e7b702b724257791/docs/figure/NYGC_single_position_ALL.Rmd/unnamed-chunk-6-1.png" target="_blank">b3ea038</a>
</td>
<td>
Andy Beck
</td>
<td>
2023-05-24
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/f77e94cfaf9173a4c9361439ca41b972e9b44c8e/docs/figure/NYGC_single_position_ALL.Rmd/unnamed-chunk-6-1.png" target="_blank">f77e94c</a>
</td>
<td>
Andy Beck
</td>
<td>
2023-05-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggsave(paste0(plot_dir, &quot;fig1_opt2c.tiff&quot;), dpi = plot_dpi, height = 7, width = 5)</code></pre>
<p>“To-From” approach:</p>
<pre class="r"><code>df &lt;- df %&gt;%
  mutate(to_n = ifelse(str_starts(subtype, &quot;A&quot;), &quot;A&quot;, ifelse(str_starts(subtype, &quot;cpg&quot;), &quot;CpG&quot;, &quot;C&quot;)),
         from_n = ifelse(str_starts(subtype, &quot;A&quot;), str_sub(subtype, 4, 4), str_sub(subtype, -1)))
df %&gt;%
  filter(abs(offset) &lt; 11) %&gt;%
  ggplot(aes(x = offset, y = re)) +
  geom_point() +
  geom_line() +
  theme_bw() +
  facet_grid(vars(from_n), vars(to_n)) +
  xlab(&quot;Relative Position&quot;) +
  ylab(&quot;Relative Entropy&quot;)</code></pre>
<p><img src="figure/NYGC_single_position_ALL.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-1">
Past versions of unnamed-chunk-7-1.png
</button>
</p>
<div id="fig-unnamed-chunk-7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/457d3ee938b4ef56dd57aad06102d34f129b2311/docs/figure/NYGC_single_position_ALL.Rmd/unnamed-chunk-7-1.png" target="_blank">457d3ee</a>
</td>
<td>
Andy Beck
</td>
<td>
2023-05-31
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/b3ea038581eafb29e23eb842e7b702b724257791/docs/figure/NYGC_single_position_ALL.Rmd/unnamed-chunk-7-1.png" target="_blank">b3ea038</a>
</td>
<td>
Andy Beck
</td>
<td>
2023-05-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggsave(paste0(plot_dir, &quot;fig1_opt3.tiff&quot;), dpi = plot_dpi, height = 5, width = 5)</code></pre>
</div>
<div id="a-c" class="section level3">
<h3>A &gt; C</h3>
<p>First let’s take a look at the overall statistic at each position for
the A &gt; C sub type. We’ll limit our initial examination to the +/- 10
bp window centered at the focal position:</p>
<pre class="r"><code>df &lt;- read_subtype_pop_res(&quot;ALL&quot;, &quot;AT_CG&quot;)

df %&gt;%
  filter(abs(offset) &lt;= 10) %&gt;%
  ggplot(aes(x = offset, y = re)) + 
  geom_point() + 
  geom_line() +
  theme_classic(base_size = 24/.pt) +
  theme(text = element_text(family = &quot;Times New Roman&quot;)) +
  xlab(&quot;Relative Position&quot;) +
  ylab(&quot;Relative Entropy&quot;)</code></pre>
<p><img src="figure/NYGC_single_position_ALL.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-1">
Past versions of unnamed-chunk-8-1.png
</button>
</p>
<div id="fig-unnamed-chunk-8-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/b3ea038581eafb29e23eb842e7b702b724257791/docs/figure/NYGC_single_position_ALL.Rmd/unnamed-chunk-8-1.png" target="_blank">b3ea038</a>
</td>
<td>
Andy Beck
</td>
<td>
2023-05-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>For this subtype, we observe that the most influential position
appears to be at the -1. We also note that, while in general the
statistic decreases as we move further from the focal position, this
decrease is not monotonic. For example, we see a higher statistic at the
-3 position than we do at the -2, and similarly a higher value is seen
at +2 than is observed at +1.</p>
<p>The statistic at each position can be decomposed into the
contributions from each of the four nucleotides. Here, a positive
residual indicates an enrichment in the singletons relative to the
expectation under the null, while a negative value indicates a
depletion.</p>
<pre class="r"><code>df_resid &lt;- read_subtype_pop_range_resid(&quot;ALL&quot;, &quot;AT_CG&quot;, size = 10)  %&gt;%
  type_convert()</code></pre>
<pre><code>
── Column specification ────────────────────────────────────────────────────────
cols(
  nuc = col_character(),
  status = col_character(),
  pos = col_double()
)</code></pre>
<pre class="r"><code>df_resid &lt;- df_resid %&gt;%
  inner_join({df %&gt;% select(offset, re)}, by = c(&quot;pos&quot; = &quot;offset&quot;))

df_resid %&gt;% 
  filter(status == &quot;singletons&quot;) %&gt;%
  select(nuc, res, re, pos) %&gt;%
  group_by(pos) %&gt;%
  mutate(pct = abs(res) / sum(abs(res))) %&gt;%
  mutate(re_cont = re * pct * sign(res)) %&gt;%
  ggplot(aes(x = pos, y = re_cont, color = nuc)) +
  geom_point() +
  geom_line() +
  xlab(&quot;Relative Position&quot;) +
  ylab(&quot;Relative Entropy&quot;) +
  guides(color = guide_legend(title = &quot;Nucleotide&quot;)) +
  theme_classic() +
  theme(text = element_text(family = &quot;Times New Roman&quot;),
        legend.position = c(.85, .8))</code></pre>
<p><img src="figure/NYGC_single_position_ALL.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-1">
Past versions of unnamed-chunk-9-1.png
</button>
</p>
<div id="fig-unnamed-chunk-9-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/b3ea038581eafb29e23eb842e7b702b724257791/docs/figure/NYGC_single_position_ALL.Rmd/unnamed-chunk-9-1.png" target="_blank">b3ea038</a>
</td>
<td>
Andy Beck
</td>
<td>
2023-05-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>For this subtype, we observe that the primary contributors to the
signal at the -1 position is a depletion of G in the singletons, while
there is an enrichment of Cs.</p>
<p>Finally, let’s look at the individual position statistics at all
positions in the +/- 1000bp window:</p>
<pre class="r"><code>df %&gt;%
  mutate(is_sig = dev &gt; qchisq(p=.05, df=3, lower.tail=FALSE)) %&gt;%
  ggplot(aes(x = offset, y = re, color = is_sig)) +
  geom_point() +
  scale_y_log10() +
  xlab(&quot;Relative Position&quot;) +
  ylab(&quot;Log10 Relative Entropy&quot;) +
  guides(color = guide_legend(title = &quot;Significant&quot;)) +
  theme_classic() +
  theme(text = element_text(family = &quot;Times New Roman&quot;),
        legend.position = c(.85, .8))</code></pre>
<p><img src="figure/NYGC_single_position_ALL.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-10-1">
Past versions of unnamed-chunk-10-1.png
</button>
</p>
<div id="fig-unnamed-chunk-10-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/b3ea038581eafb29e23eb842e7b702b724257791/docs/figure/NYGC_single_position_ALL.Rmd/unnamed-chunk-10-1.png" target="_blank">b3ea038</a>
</td>
<td>
Andy Beck
</td>
<td>
2023-05-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="a-g" class="section level3">
<h3>A &gt; G</h3>
<p>Let’s take a look at the A &gt; G subtype:</p>
<pre class="r"><code>df &lt;- read_subtype_pop_res(&quot;ALL&quot;, &quot;AT_GC&quot;)

df %&gt;%
  filter(abs(offset) &lt;= 10) %&gt;%
  ggplot(aes(x = offset, y = re)) + 
  geom_point() + 
  geom_line() +
  theme_classic(base_size = 24/.pt) +
  theme(text = element_text(family = &quot;Times New Roman&quot;)) +
  xlab(&quot;Relative Position&quot;) +
  ylab(&quot;Relative Entropy&quot;)</code></pre>
<p><img src="figure/NYGC_single_position_ALL.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-1">
Past versions of unnamed-chunk-11-1.png
</button>
</p>
<div id="fig-unnamed-chunk-11-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/b3ea038581eafb29e23eb842e7b702b724257791/docs/figure/NYGC_single_position_ALL.Rmd/unnamed-chunk-11-1.png" target="_blank">b3ea038</a>
</td>
<td>
Andy Beck
</td>
<td>
2023-05-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Here, we observe that the most influential position is the +1
position. Let’s look and see what the nucleotide-level contributions to
each individual statistic looks like:</p>
<pre class="r"><code>df_resid &lt;- read_subtype_pop_range_resid(&quot;ALL&quot;, &quot;AT_GC&quot;, size = 10)  %&gt;%
  type_convert()</code></pre>
<pre><code>
── Column specification ────────────────────────────────────────────────────────
cols(
  nuc = col_character(),
  status = col_character(),
  pos = col_double()
)</code></pre>
<pre class="r"><code>df_resid &lt;- df_resid %&gt;%
  inner_join({df %&gt;% select(offset, re)}, by = c(&quot;pos&quot; = &quot;offset&quot;))

df_resid %&gt;% 
  filter(status == &quot;singletons&quot;) %&gt;%
  select(nuc, res, re, pos) %&gt;%
  group_by(pos) %&gt;%
  mutate(pct = abs(res) / sum(abs(res))) %&gt;%
  mutate(re_cont = re * pct * sign(res)) %&gt;%
  ggplot(aes(x = pos, y = re_cont, color = nuc)) +
  geom_point() +
  geom_line(show.legend = FALSE) +
  xlab(&quot;Relative Position&quot;) +
  ylab(&quot;Relative Entropy&quot;) +
  guides(color = guide_legend(title = &quot;Nucleotide&quot;)) +
  theme_classic() +
  theme(text = element_text(family = &quot;Times New Roman&quot;),
        legend.position = c(.85, .8))</code></pre>
<p><img src="figure/NYGC_single_position_ALL.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-1">
Past versions of unnamed-chunk-12-1.png
</button>
</p>
<div id="fig-unnamed-chunk-12-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/457d3ee938b4ef56dd57aad06102d34f129b2311/docs/figure/NYGC_single_position_ALL.Rmd/unnamed-chunk-12-1.png" target="_blank">457d3ee</a>
</td>
<td>
Andy Beck
</td>
<td>
2023-05-31
</td>
</tr>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/b3ea038581eafb29e23eb842e7b702b724257791/docs/figure/NYGC_single_position_ALL.Rmd/unnamed-chunk-12-1.png" target="_blank">b3ea038</a>
</td>
<td>
Andy Beck
</td>
<td>
2023-05-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>For this subtype, we see that the signal at the +1 position is driven
by an enrichment of T and a depletion of A. Also notice that at -1, the
primary driver is an enrichment of C; we also observed an enrichment of
C in the A&gt;C subtype.</p>
<p>Let’s also plot the individual position statistics up to +/- 1000bp
from the focal position. Here we’ll color significance based on the null
distribution being a chi-square with df = 3</p>
<pre class="r"><code>df %&gt;%
  mutate(is_sig = dev &gt; qchisq(p=.05, df=3, lower.tail=FALSE)) %&gt;%
  ggplot(aes(x = offset, y = re, color = is_sig)) +
  geom_point() +
  scale_y_log10() +
  xlab(&quot;Relative Position&quot;) +
  ylab(&quot;Log10 Relative Entropy&quot;) +
  guides(color = guide_legend(title = &quot;Significant&quot;)) +
  theme_classic() +
  theme(text = element_text(family = &quot;Times New Roman&quot;),
        legend.position = c(.85, .8))</code></pre>
<p><img src="figure/NYGC_single_position_ALL.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-13-1">
Past versions of unnamed-chunk-13-1.png
</button>
</p>
<div id="fig-unnamed-chunk-13-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/b3ea038581eafb29e23eb842e7b702b724257791/docs/figure/NYGC_single_position_ALL.Rmd/unnamed-chunk-13-1.png" target="_blank">b3ea038</a>
</td>
<td>
Andy Beck
</td>
<td>
2023-05-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="a-t" class="section level3">
<h3>A &gt; T</h3>
<pre class="r"><code>df &lt;- read_subtype_pop_res(&quot;ALL&quot;, &quot;AT_TA&quot;)

df %&gt;%
  filter(abs(offset) &lt;= 10) %&gt;%
  ggplot(aes(x = offset, y = re)) + 
  geom_point() + 
  geom_line() +
  theme_classic(base_size = 24/.pt) +
  theme(text = element_text(family = &quot;Times New Roman&quot;)) +
  xlab(&quot;Relative Position&quot;) +
  ylab(&quot;Relative Entropy&quot;)</code></pre>
<p><img src="figure/NYGC_single_position_ALL.Rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-1">
Past versions of unnamed-chunk-14-1.png
</button>
</p>
<div id="fig-unnamed-chunk-14-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/457d3ee938b4ef56dd57aad06102d34f129b2311/docs/figure/NYGC_single_position_ALL.Rmd/unnamed-chunk-14-1.png" target="_blank">457d3ee</a>
</td>
<td>
Andy Beck
</td>
<td>
2023-05-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>For this sub type we observe the strongest signal at the +1 relative
position. Below we observe that this signal is driven by an enrichment
of T and depletion of A and G. Also note that we also have an enrichment
of C at the +1, although this is far more mild than observed in the
previous two subtypes.</p>
<pre class="r"><code>df_resid &lt;- read_subtype_pop_range_resid(&quot;ALL&quot;, &quot;AT_TA&quot;, size = 10)  %&gt;%
  type_convert()</code></pre>
<pre><code>
── Column specification ────────────────────────────────────────────────────────
cols(
  nuc = col_character(),
  status = col_character(),
  pos = col_double()
)</code></pre>
<pre class="r"><code>df_resid &lt;- df_resid %&gt;%
  inner_join({df %&gt;% select(offset, re)}, by = c(&quot;pos&quot; = &quot;offset&quot;))

df_resid %&gt;% 
  filter(status == &quot;singletons&quot;) %&gt;%
  select(nuc, res, re, pos) %&gt;%
  group_by(pos) %&gt;%
  mutate(pct = abs(res) / sum(abs(res))) %&gt;%
  mutate(re_cont = re * pct * sign(res)) %&gt;%
  ggplot(aes(x = pos, y = re_cont, color = nuc)) +
  geom_point() +
  geom_line(show.legend = FALSE) +
  xlab(&quot;Relative Position&quot;) +
  ylab(&quot;Relative Entropy&quot;) +
  guides(color = guide_legend(title = &quot;Nucleotide&quot;)) +
  theme_classic() +
  theme(text = element_text(family = &quot;Times New Roman&quot;),
        legend.position = c(.85, .8))</code></pre>
<p><img src="figure/NYGC_single_position_ALL.Rmd/unnamed-chunk-15-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-15-1">
Past versions of unnamed-chunk-15-1.png
</button>
</p>
<div id="fig-unnamed-chunk-15-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/457d3ee938b4ef56dd57aad06102d34f129b2311/docs/figure/NYGC_single_position_ALL.Rmd/unnamed-chunk-15-1.png" target="_blank">457d3ee</a>
</td>
<td>
Andy Beck
</td>
<td>
2023-05-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>df %&gt;%
  mutate(is_sig = dev &gt; qchisq(p=.05, df=3, lower.tail=FALSE)) %&gt;%
  ggplot(aes(x = offset, y = re, color = is_sig)) +
  geom_point() +
  scale_y_log10() +
  xlab(&quot;Relative Position&quot;) +
  ylab(&quot;Log10 Relative Entropy&quot;) +
  guides(color = guide_legend(title = &quot;Significant&quot;)) +
  theme_classic() +
  theme(text = element_text(family = &quot;Times New Roman&quot;),
        legend.position = c(.85, .8))</code></pre>
<p><img src="figure/NYGC_single_position_ALL.Rmd/unnamed-chunk-16-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-16-1">
Past versions of unnamed-chunk-16-1.png
</button>
</p>
<div id="fig-unnamed-chunk-16-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/457d3ee938b4ef56dd57aad06102d34f129b2311/docs/figure/NYGC_single_position_ALL.Rmd/unnamed-chunk-16-1.png" target="_blank">457d3ee</a>
</td>
<td>
Andy Beck
</td>
<td>
2023-05-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="c-a" class="section level3">
<h3>C &gt; A</h3>
<pre class="r"><code>df &lt;- read_subtype_pop_res(&quot;ALL&quot;, &quot;GC_TA&quot;)

df %&gt;%
  filter(abs(offset) &lt;= 10) %&gt;%
  ggplot(aes(x = offset, y = re)) + 
  geom_point() + 
  geom_line() +
  theme_classic(base_size = 24/.pt) +
  theme(text = element_text(family = &quot;Times New Roman&quot;)) +
  xlab(&quot;Relative Position&quot;) +
  ylab(&quot;Relative Entropy&quot;)</code></pre>
<p><img src="figure/NYGC_single_position_ALL.Rmd/unnamed-chunk-17-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-17-1">
Past versions of unnamed-chunk-17-1.png
</button>
</p>
<div id="fig-unnamed-chunk-17-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/457d3ee938b4ef56dd57aad06102d34f129b2311/docs/figure/NYGC_single_position_ALL.Rmd/unnamed-chunk-17-1.png" target="_blank">457d3ee</a>
</td>
<td>
Andy Beck
</td>
<td>
2023-05-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>For this subtype, we observe the strongest signal at the +1 position.
Note here that we have conditioned on the nucleotide at +1 not being a
G.</p>
<pre class="r"><code>df_resid &lt;- read_subtype_pop_range_resid(&quot;ALL&quot;, &quot;GC_TA&quot;, size = 10)  %&gt;%
  type_convert()</code></pre>
<pre><code>
── Column specification ────────────────────────────────────────────────────────
cols(
  nuc = col_character(),
  status = col_character(),
  pos = col_double()
)</code></pre>
<pre class="r"><code>df_resid &lt;- df_resid %&gt;%
  inner_join({df %&gt;% select(offset, re)}, by = c(&quot;pos&quot; = &quot;offset&quot;))

df_resid %&gt;% 
  filter(status == &quot;singletons&quot;) %&gt;%
  select(nuc, res, re, pos) %&gt;%
  group_by(pos) %&gt;%
  mutate(pct = abs(res) / sum(abs(res))) %&gt;%
  mutate(re_cont = re * pct * sign(res)) %&gt;%
  ggplot(aes(x = pos, y = re_cont, color = nuc)) +
  geom_point() +
  geom_line(show.legend = FALSE) +
  xlab(&quot;Relative Position&quot;) +
  ylab(&quot;Relative Entropy&quot;) +
  guides(color = guide_legend(title = &quot;Nucleotide&quot;)) +
  theme_classic() +
  theme(text = element_text(family = &quot;Times New Roman&quot;),
        legend.position = c(.85, .8))</code></pre>
<p><img src="figure/NYGC_single_position_ALL.Rmd/unnamed-chunk-18-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-18-1">
Past versions of unnamed-chunk-18-1.png
</button>
</p>
<div id="fig-unnamed-chunk-18-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/457d3ee938b4ef56dd57aad06102d34f129b2311/docs/figure/NYGC_single_position_ALL.Rmd/unnamed-chunk-18-1.png" target="_blank">457d3ee</a>
</td>
<td>
Andy Beck
</td>
<td>
2023-05-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>And just to be absolutely sure that none of our singletons or
controls have a G at the +1 flanking position:</p>
<pre class="r"><code>df_resid %&gt;% filter(pos == 1) %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">nuc</th>
<th align="left">status</th>
<th align="right">n</th>
<th align="right">fitted</th>
<th align="right">res</th>
<th align="right">pos</th>
<th align="right">re</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">A</td>
<td align="left">singletons</td>
<td align="right">1757786</td>
<td align="right">1889785</td>
<td align="right">-97.1722737</td>
<td align="right">1</td>
<td align="right">0.0037099</td>
</tr>
<tr class="even">
<td align="left">C</td>
<td align="left">singletons</td>
<td align="right">1753906</td>
<td align="right">1330278</td>
<td align="right">350.0058837</td>
<td align="right">1</td>
<td align="right">0.0037099</td>
</tr>
<tr class="odd">
<td align="left">G</td>
<td align="left">singletons</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">-0.0000312</td>
<td align="right">1</td>
<td align="right">0.0037099</td>
</tr>
<tr class="even">
<td align="left">T</td>
<td align="left">singletons</td>
<td align="right">1473853</td>
<td align="right">1765482</td>
<td align="right">-225.9826183</td>
<td align="right">1</td>
<td align="right">0.0037099</td>
</tr>
<tr class="odd">
<td align="left">A</td>
<td align="left">controls</td>
<td align="right">9580924</td>
<td align="right">9448925</td>
<td align="right">42.8423172</td>
<td align="right">1</td>
<td align="right">0.0037099</td>
</tr>
<tr class="even">
<td align="left">C</td>
<td align="left">controls</td>
<td align="right">6227765</td>
<td align="right">6651392</td>
<td align="right">-166.0500778</td>
<td align="right">1</td>
<td align="right">0.0037099</td>
</tr>
<tr class="odd">
<td align="left">G</td>
<td align="left">controls</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">-0.0000697</td>
<td align="right">1</td>
<td align="right">0.0037099</td>
</tr>
<tr class="even">
<td align="left">T</td>
<td align="left">controls</td>
<td align="right">9119036</td>
<td align="right">8827407</td>
<td align="right">97.6220666</td>
<td align="right">1</td>
<td align="right">0.0037099</td>
</tr>
</tbody>
</table>
<pre class="r"><code>df %&gt;%
  mutate(is_sig = dev &gt; qchisq(p=.05, df=3, lower.tail=FALSE)) %&gt;%
  ggplot(aes(x = offset, y = re, color = is_sig)) +
  geom_point() +
  scale_y_log10() +
  xlab(&quot;Relative Position&quot;) +
  ylab(&quot;Log10 Relative Entropy&quot;) +
  guides(color = guide_legend(title = &quot;Significant&quot;)) +
  theme_classic() +
  theme(text = element_text(family = &quot;Times New Roman&quot;),
        legend.position = c(.85, .8))</code></pre>
<p><img src="figure/NYGC_single_position_ALL.Rmd/unnamed-chunk-20-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-20-1">
Past versions of unnamed-chunk-20-1.png
</button>
</p>
<div id="fig-unnamed-chunk-20-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/457d3ee938b4ef56dd57aad06102d34f129b2311/docs/figure/NYGC_single_position_ALL.Rmd/unnamed-chunk-20-1.png" target="_blank">457d3ee</a>
</td>
<td>
Andy Beck
</td>
<td>
2023-05-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="c-t" class="section level3">
<h3>C &gt; T</h3>
<pre class="r"><code>df &lt;- read_subtype_pop_res(&quot;ALL&quot;, &quot;GC_AT&quot;)

df %&gt;%
  filter(abs(offset) &lt;= 10) %&gt;%
  ggplot(aes(x = offset, y = re)) + 
  geom_point() + 
  geom_line() +
  theme_classic(base_size = 24/.pt) +
  theme(text = element_text(family = &quot;Times New Roman&quot;)) +
  xlab(&quot;Relative Position&quot;) +
  ylab(&quot;Relative Entropy&quot;)</code></pre>
<p><img src="figure/NYGC_single_position_ALL.Rmd/unnamed-chunk-21-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-21-1">
Past versions of unnamed-chunk-21-1.png
</button>
</p>
<div id="fig-unnamed-chunk-21-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/04837fdc1d6e457e8e4b0832bdc731e79954c9f4/docs/figure/NYGC_single_position_ALL.Rmd/unnamed-chunk-21-1.png" target="_blank">04837fd</a>
</td>
<td>
Andy Beck
</td>
<td>
2023-05-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Here, for this subtype we observe the strongest signals at -1 and +2.
Note again that we have conditioned on this not being at CpG sites.
Below we see that the primary driver of the signal is a depletion of T
and an enrichment of A at the -1.</p>
<pre class="r"><code>df_resid &lt;- read_subtype_pop_range_resid(&quot;ALL&quot;, &quot;GC_AT&quot;, size = 10)  %&gt;%
  type_convert()</code></pre>
<pre><code>
── Column specification ────────────────────────────────────────────────────────
cols(
  nuc = col_character(),
  status = col_character(),
  pos = col_double()
)</code></pre>
<pre class="r"><code>df_resid &lt;- df_resid %&gt;%
  inner_join({df %&gt;% select(offset, re)}, by = c(&quot;pos&quot; = &quot;offset&quot;))

df_resid %&gt;% 
  filter(status == &quot;singletons&quot;) %&gt;%
  select(nuc, res, re, pos) %&gt;%
  group_by(pos) %&gt;%
  mutate(pct = abs(res) / sum(abs(res))) %&gt;%
  mutate(re_cont = re * pct * sign(res)) %&gt;%
  ggplot(aes(x = pos, y = re_cont, color = nuc)) +
  geom_point() +
  geom_line(show.legend = FALSE) +
  xlab(&quot;Relative Position&quot;) +
  ylab(&quot;Relative Entropy&quot;) +
  guides(color = guide_legend(title = &quot;Nucleotide&quot;)) +
  theme_classic() +
  theme(text = element_text(family = &quot;Times New Roman&quot;),
        legend.position = c(.85, .8))</code></pre>
<p><img src="figure/NYGC_single_position_ALL.Rmd/unnamed-chunk-22-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-22-1">
Past versions of unnamed-chunk-22-1.png
</button>
</p>
<div id="fig-unnamed-chunk-22-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/04837fdc1d6e457e8e4b0832bdc731e79954c9f4/docs/figure/NYGC_single_position_ALL.Rmd/unnamed-chunk-22-1.png" target="_blank">04837fd</a>
</td>
<td>
Andy Beck
</td>
<td>
2023-05-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>df %&gt;%
  mutate(is_sig = dev &gt; qchisq(p=.05, df=3, lower.tail=FALSE)) %&gt;%
  ggplot(aes(x = offset, y = re, color = is_sig)) +
  geom_point() +
  scale_y_log10() +
  xlab(&quot;Relative Position&quot;) +
  ylab(&quot;Log10 Relative Entropy&quot;) +
  guides(color = guide_legend(title = &quot;Significant&quot;)) +
  theme_classic() +
  theme(text = element_text(family = &quot;Times New Roman&quot;),
        legend.position = c(.85, .8))</code></pre>
<p><img src="figure/NYGC_single_position_ALL.Rmd/unnamed-chunk-23-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-23-1">
Past versions of unnamed-chunk-23-1.png
</button>
</p>
<div id="fig-unnamed-chunk-23-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/04837fdc1d6e457e8e4b0832bdc731e79954c9f4/docs/figure/NYGC_single_position_ALL.Rmd/unnamed-chunk-23-1.png" target="_blank">04837fd</a>
</td>
<td>
Andy Beck
</td>
<td>
2023-05-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="c-g" class="section level3">
<h3>C &gt; G</h3>
<pre class="r"><code>df &lt;- read_subtype_pop_res(&quot;ALL&quot;, &quot;GC_CG&quot;)

df %&gt;%
  filter(abs(offset) &lt;= 10) %&gt;%
  ggplot(aes(x = offset, y = re)) + 
  geom_point() + 
  geom_line() +
  theme_classic(base_size = 24/.pt) +
  theme(text = element_text(family = &quot;Times New Roman&quot;)) +
  xlab(&quot;Relative Position&quot;) +
  ylab(&quot;Relative Entropy&quot;)</code></pre>
<p><img src="figure/NYGC_single_position_ALL.Rmd/unnamed-chunk-24-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-24-1">
Past versions of unnamed-chunk-24-1.png
</button>
</p>
<div id="fig-unnamed-chunk-24-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/04837fdc1d6e457e8e4b0832bdc731e79954c9f4/docs/figure/NYGC_single_position_ALL.Rmd/unnamed-chunk-24-1.png" target="_blank">04837fd</a>
</td>
<td>
Andy Beck
</td>
<td>
2023-05-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>df_resid &lt;- read_subtype_pop_range_resid(&quot;ALL&quot;, &quot;GC_CG&quot;, size = 10)  %&gt;%
  type_convert()</code></pre>
<pre><code>
── Column specification ────────────────────────────────────────────────────────
cols(
  nuc = col_character(),
  status = col_character(),
  pos = col_double()
)</code></pre>
<pre class="r"><code>df_resid &lt;- df_resid %&gt;%
  inner_join({df %&gt;% select(offset, re)}, by = c(&quot;pos&quot; = &quot;offset&quot;))

df_resid %&gt;% 
  filter(status == &quot;singletons&quot;) %&gt;%
  select(nuc, res, re, pos) %&gt;%
  group_by(pos) %&gt;%
  mutate(pct = abs(res) / sum(abs(res))) %&gt;%
  mutate(re_cont = re * pct * sign(res)) %&gt;%
  ggplot(aes(x = pos, y = re_cont, color = nuc)) +
  geom_point() +
  geom_line(show.legend = FALSE) +
  xlab(&quot;Relative Position&quot;) +
  ylab(&quot;Relative Entropy&quot;) +
  guides(color = guide_legend(title = &quot;Nucleotide&quot;)) +
  theme_classic() +
  theme(text = element_text(family = &quot;Times New Roman&quot;),
        legend.position = c(.85, .8))</code></pre>
<p><img src="figure/NYGC_single_position_ALL.Rmd/unnamed-chunk-25-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>df_resid %&gt;% filter(pos == 1) %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">nuc</th>
<th align="left">status</th>
<th align="right">n</th>
<th align="right">fitted</th>
<th align="right">res</th>
<th align="right">pos</th>
<th align="right">re</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">A</td>
<td align="left">singletons</td>
<td align="right">1359622</td>
<td align="right">1637030</td>
<td align="right">-223.4169533</td>
<td align="right">1</td>
<td align="right">0.0017335</td>
</tr>
<tr class="even">
<td align="left">C</td>
<td align="left">singletons</td>
<td align="right">1236987</td>
<td align="right">1167028</td>
<td align="right">64.1277391</td>
<td align="right">1</td>
<td align="right">0.0017335</td>
</tr>
<tr class="odd">
<td align="left">G</td>
<td align="left">singletons</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">-0.0000296</td>
<td align="right">1</td>
<td align="right">0.0017335</td>
</tr>
<tr class="even">
<td align="left">T</td>
<td align="left">singletons</td>
<td align="right">1905337</td>
<td align="right">1697888</td>
<td align="right">156.1179015</td>
<td align="right">1</td>
<td align="right">0.0017335</td>
</tr>
<tr class="odd">
<td align="left">A</td>
<td align="left">controls</td>
<td align="right">8462557</td>
<td align="right">8185149</td>
<td align="right">96.4227489</td>
<td align="right">1</td>
<td align="right">0.0017335</td>
</tr>
<tr class="even">
<td align="left">C</td>
<td align="left">controls</td>
<td align="right">5765183</td>
<td align="right">5835142</td>
<td align="right">-29.0193124</td>
<td align="right">1</td>
<td align="right">0.0017335</td>
</tr>
<tr class="odd">
<td align="left">G</td>
<td align="left">controls</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">-0.0000662</td>
<td align="right">1</td>
<td align="right">0.0017335</td>
</tr>
<tr class="even">
<td align="left">T</td>
<td align="left">controls</td>
<td align="right">8281990</td>
<td align="right">8489439</td>
<td align="right">-71.4916847</td>
<td align="right">1</td>
<td align="right">0.0017335</td>
</tr>
</tbody>
</table>
<pre class="r"><code>df %&gt;%
  mutate(is_sig = dev &gt; qchisq(p=.05, df=3, lower.tail=FALSE)) %&gt;%
  ggplot(aes(x = offset, y = re, color = is_sig)) +
  geom_point() +
  scale_y_log10() +
  xlab(&quot;Relative Position&quot;) +
  ylab(&quot;Log10 Relative Entropy&quot;) +
  guides(color = guide_legend(title = &quot;Significant&quot;)) +
  theme_classic() +
  theme(text = element_text(family = &quot;Times New Roman&quot;),
        legend.position = c(.85, .8))</code></pre>
<p><img src="figure/NYGC_single_position_ALL.Rmd/unnamed-chunk-27-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-27-1">
Past versions of unnamed-chunk-27-1.png
</button>
</p>
<div id="fig-unnamed-chunk-27-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/04837fdc1d6e457e8e4b0832bdc731e79954c9f4/docs/figure/NYGC_single_position_ALL.Rmd/unnamed-chunk-27-1.png" target="_blank">04837fd</a>
</td>
<td>
Andy Beck
</td>
<td>
2023-05-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cpg-to-apg" class="section level3">
<h3>CpG to ApG</h3>
<pre class="r"><code>df &lt;- read_subtype_pop_res(&quot;ALL&quot;, &quot;cpg_GC_TA&quot;)

df %&gt;%
  filter(abs(offset) &lt;= 10) %&gt;%
  ggplot(aes(x = offset, y = re)) + 
  geom_point() + 
  geom_line() +
  theme_classic(base_size = 24/.pt) +
  theme(text = element_text(family = &quot;Times New Roman&quot;)) +
  xlab(&quot;Relative Position&quot;) +
  ylab(&quot;Relative Entropy&quot;)</code></pre>
<p><img src="figure/NYGC_single_position_ALL.Rmd/unnamed-chunk-28-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-28-1">
Past versions of unnamed-chunk-28-1.png
</button>
</p>
<div id="fig-unnamed-chunk-28-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/04837fdc1d6e457e8e4b0832bdc731e79954c9f4/docs/figure/NYGC_single_position_ALL.Rmd/unnamed-chunk-28-1.png" target="_blank">04837fd</a>
</td>
<td>
Andy Beck
</td>
<td>
2023-05-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>df_resid &lt;- read_subtype_pop_range_resid(&quot;ALL&quot;, &quot;cpg_GC_TA&quot;, size = 10)  %&gt;%
  type_convert()</code></pre>
<pre><code>
── Column specification ────────────────────────────────────────────────────────
cols(
  nuc = col_character(),
  status = col_character(),
  pos = col_double()
)</code></pre>
<pre class="r"><code>df_resid &lt;- df_resid %&gt;%
  inner_join({df %&gt;% select(offset, re)}, by = c(&quot;pos&quot; = &quot;offset&quot;))

df_resid %&gt;% 
  filter(status == &quot;singletons&quot;) %&gt;%
  select(nuc, res, re, pos) %&gt;%
  group_by(pos) %&gt;%
  mutate(pct = abs(res) / sum(abs(res))) %&gt;%
  mutate(re_cont = re * pct * sign(res)) %&gt;%
  ggplot(aes(x = pos, y = re_cont, color = nuc)) +
  geom_point() +
  geom_line(show.legend = FALSE) +
  xlab(&quot;Relative Position&quot;) +
  ylab(&quot;Relative Entropy&quot;) +
  guides(color = guide_legend(title = &quot;Nucleotide&quot;)) +
  theme_classic() +
  theme(text = element_text(family = &quot;Times New Roman&quot;),
        legend.position = c(.85, .8))</code></pre>
<p><img src="figure/NYGC_single_position_ALL.Rmd/unnamed-chunk-29-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-29-1">
Past versions of unnamed-chunk-29-1.png
</button>
</p>
<div id="fig-unnamed-chunk-29-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/04837fdc1d6e457e8e4b0832bdc731e79954c9f4/docs/figure/NYGC_single_position_ALL.Rmd/unnamed-chunk-29-1.png" target="_blank">04837fd</a>
</td>
<td>
Andy Beck
</td>
<td>
2023-05-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>df %&gt;%
  mutate(is_sig = dev &gt; qchisq(p=.05, df=3, lower.tail=FALSE)) %&gt;%
  ggplot(aes(x = offset, y = re, color = is_sig)) +
  geom_point() +
  scale_y_log10() +
  xlab(&quot;Relative Position&quot;) +
  ylab(&quot;Log10 Relative Entropy&quot;) +
  guides(color = guide_legend(title = &quot;Significant&quot;)) +
  theme_classic() +
  theme(text = element_text(family = &quot;Times New Roman&quot;),
        legend.position = c(.85, .8))</code></pre>
<p><img src="figure/NYGC_single_position_ALL.Rmd/unnamed-chunk-30-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-30-1">
Past versions of unnamed-chunk-30-1.png
</button>
</p>
<div id="fig-unnamed-chunk-30-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/04837fdc1d6e457e8e4b0832bdc731e79954c9f4/docs/figure/NYGC_single_position_ALL.Rmd/unnamed-chunk-30-1.png" target="_blank">04837fd</a>
</td>
<td>
Andy Beck
</td>
<td>
2023-05-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cpg-to-tpg" class="section level3">
<h3>CpG to TpG</h3>
<pre class="r"><code>df &lt;- read_subtype_pop_res(&quot;ALL&quot;, &quot;cpg_GC_AT&quot;)

df %&gt;%
  filter(abs(offset) &lt;= 10) %&gt;%
  ggplot(aes(x = offset, y = re)) + 
  geom_point() + 
  geom_line() +
  theme_classic(base_size = 24/.pt) +
  theme(text = element_text(family = &quot;Times New Roman&quot;)) +
  xlab(&quot;Relative Position&quot;) +
  ylab(&quot;Relative Entropy&quot;)</code></pre>
<p><img src="figure/NYGC_single_position_ALL.Rmd/unnamed-chunk-31-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-31-1">
Past versions of unnamed-chunk-31-1.png
</button>
</p>
<div id="fig-unnamed-chunk-31-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/theandyb/LSCI/blob/04837fdc1d6e457e8e4b0832bdc731e79954c9f4/docs/figure/NYGC_single_position_ALL.Rmd/unnamed-chunk-31-1.png" target="_blank">04837fd</a>
</td>
<td>
Andy Beck
</td>
<td>
2023-05-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>df_resid &lt;- read_subtype_pop_range_resid(&quot;ALL&quot;, &quot;cpg_GC_AT&quot;, size = 10)  %&gt;%
  type_convert()</code></pre>
<pre><code>
── Column specification ────────────────────────────────────────────────────────
cols(
  nuc = col_character(),
  status = col_character(),
  pos = col_double()
)</code></pre>
<pre class="r"><code>df_resid &lt;- df_resid %&gt;%
  inner_join({df %&gt;% select(offset, re)}, by = c(&quot;pos&quot; = &quot;offset&quot;))

df_resid %&gt;% 
  filter(status == &quot;singletons&quot;) %&gt;%
  select(nuc, res, re, pos) %&gt;%
  group_by(pos) %&gt;%
  mutate(pct = abs(res) / sum(abs(res))) %&gt;%
  mutate(re_cont = re * pct * sign(res)) %&gt;%
  ggplot(aes(x = pos, y = re_cont, color = nuc)) +
  geom_point() +
  geom_line(show.legend = FALSE) +
  xlab(&quot;Relative Position&quot;) +
  ylab(&quot;Relative Entropy&quot;) +
  guides(color = guide_legend(title = &quot;Nucleotide&quot;)) +
  theme_classic() +
  theme(text = element_text(family = &quot;Times New Roman&quot;),
        legend.position = c(.85, .8))</code></pre>
<p><img src="figure/NYGC_single_position_ALL.Rmd/unnamed-chunk-32-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>df %&gt;%
  mutate(is_sig = dev &gt; qchisq(p=.05, df=3, lower.tail=FALSE)) %&gt;%
  ggplot(aes(x = offset, y = re, color = is_sig)) +
  geom_point() +
  scale_y_log10() +
  xlab(&quot;Relative Position&quot;) +
  ylab(&quot;Log10 Relative Entropy&quot;) +
  guides(color = guide_legend(title = &quot;Significant&quot;)) +
  theme_classic() +
  theme(text = element_text(family = &quot;Times New Roman&quot;),
        legend.position = c(.85, .8))</code></pre>
<p><img src="figure/NYGC_single_position_ALL.Rmd/unnamed-chunk-33-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="cpg-to-gpg" class="section level3">
<h3>CpG to GpG</h3>
<pre class="r"><code>df &lt;- read_subtype_pop_res(&quot;ALL&quot;, &quot;cpg_GC_CG&quot;)

df %&gt;%
  filter(abs(offset) &lt;= 10) %&gt;%
  ggplot(aes(x = offset, y = re)) + 
  geom_point() + 
  geom_line() +
  theme_classic(base_size = 24/.pt) +
  theme(text = element_text(family = &quot;Times New Roman&quot;)) +
  xlab(&quot;Relative Position&quot;) +
  ylab(&quot;Relative Entropy&quot;)</code></pre>
<p><img src="figure/NYGC_single_position_ALL.Rmd/unnamed-chunk-34-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>df_resid &lt;- read_subtype_pop_range_resid(&quot;ALL&quot;, &quot;cpg_GC_CG&quot;, size = 10)  %&gt;%
  type_convert()</code></pre>
<pre><code>
── Column specification ────────────────────────────────────────────────────────
cols(
  nuc = col_character(),
  status = col_character(),
  pos = col_double()
)</code></pre>
<pre class="r"><code>df_resid &lt;- df_resid %&gt;%
  inner_join({df %&gt;% select(offset, re)}, by = c(&quot;pos&quot; = &quot;offset&quot;))

df_resid %&gt;% 
  filter(status == &quot;singletons&quot;) %&gt;%
  select(nuc, res, re, pos) %&gt;%
  group_by(pos) %&gt;%
  mutate(pct = abs(res) / sum(abs(res))) %&gt;%
  mutate(re_cont = re * pct * sign(res)) %&gt;%
  ggplot(aes(x = pos, y = re_cont, color = nuc)) +
  geom_point() +
  geom_line(show.legend = FALSE) +
  xlab(&quot;Relative Position&quot;) +
  ylab(&quot;Relative Entropy&quot;) +
  guides(color = guide_legend(title = &quot;Nucleotide&quot;)) +
  theme_classic() +
  theme(text = element_text(family = &quot;Times New Roman&quot;),
        legend.position = c(.85, 0.26))</code></pre>
<p><img src="figure/NYGC_single_position_ALL.Rmd/unnamed-chunk-35-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>df %&gt;%
  mutate(is_sig = dev &gt; qchisq(p=.05, df=3, lower.tail=FALSE)) %&gt;%
  ggplot(aes(x = offset, y = re, color = is_sig)) +
  geom_point() +
  scale_y_log10() +
  xlab(&quot;Relative Position&quot;) +
  ylab(&quot;Log10 Relative Entropy&quot;) +
  guides(color = guide_legend(title = &quot;Significant&quot;)) +
  theme_classic() +
  theme(text = element_text(family = &quot;Times New Roman&quot;),
        legend.position = c(.85, .8))</code></pre>
<p><img src="figure/NYGC_single_position_ALL.Rmd/unnamed-chunk-36-1.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.0 (2023-04-21)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 18.04.6 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/openblas/libblas.so.3 
LAPACK: /usr/lib/x86_64-linux-gnu/libopenblasp-r0.2.20.so;  LAPACK version 3.7.1

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

time zone: America/Detroit
tzcode source: system (glibc)

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] egg_0.4.5       gridExtra_2.3   patchwork_1.1.1 lubridate_1.9.2
 [5] forcats_1.0.0   stringr_1.5.0   dplyr_1.1.2     purrr_1.0.1    
 [9] readr_2.1.4     tidyr_1.3.0     tibble_3.2.1    ggplot2_3.4.2  
[13] tidyverse_2.0.0 workflowr_1.7.0

loaded via a namespace (and not attached):
 [1] gtable_0.3.0      xfun_0.39         bslib_0.3.1       processx_3.8.1   
 [5] callr_3.7.3       tzdb_0.4.0        vctrs_0.6.2       tools_4.3.0      
 [9] ps_1.7.0          generics_0.1.3    parallel_4.3.0    fansi_1.0.4      
[13] highr_0.9         pkgconfig_2.0.3   lifecycle_1.0.3   compiler_4.3.0   
[17] farver_2.1.0      git2r_0.32.0      textshaping_0.3.6 munsell_0.5.0    
[21] getPass_0.2-2     httpuv_1.6.11     htmltools_0.5.5   sass_0.4.1       
[25] yaml_2.3.7        later_1.3.0       pillar_1.9.0      crayon_1.5.2     
[29] jquerylib_0.1.4   whisker_0.4       tidyselect_1.2.0  digest_0.6.31    
[33] stringi_1.7.12    labeling_0.4.2    rprojroot_2.0.3   fastmap_1.1.0    
[37] grid_4.3.0        colorspace_2.0-3  cli_3.6.1         magrittr_2.0.3   
[41] utf8_1.2.2        withr_2.5.0       scales_1.2.0      promises_1.2.0.1 
[45] bit64_4.0.5       timechange_0.2.0  rmarkdown_2.14    httr_1.4.6       
[49] bit_4.0.4         ragg_1.2.5        hms_1.1.3         evaluate_0.15    
[53] knitr_1.42        rlang_1.1.1       Rcpp_1.0.10       glue_1.6.2       
[57] rstudioapi_0.14   vroom_1.6.3       jsonlite_1.8.4    R6_2.5.1         
[61] systemfonts_1.0.4 fs_1.6.2         </code></pre>
</div>
</div>
</div>
</div>

<div id="disqus_thread"></div>
<script>
(function() { // DON'T EDIT BELOW THIS LINE
var d = document, s = d.createElement('script');
s.src = 'https://lsci-doc.disqus.com/embed.js';
s.setAttribute('data-timestamp', +new Date());
(d.head || d.body).appendChild(s);
})();
</script>
<noscript>Please enable JavaScript to view the
<a href="https://disqus.com/?ref_noscript">
  comments powered by Disqus.</a></noscript>

<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
