<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Andy Beck" />

<meta name="date" content="2021-11-01" />

<title>both_outline</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Local Sequence Context Influence</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/theandyb/LSCI">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">both_outline</h1>
<h4 class="author">Andy Beck</h4>
<h4 class="date">2021-11-01</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-11-10
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 2 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>LSCI/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomtheandybLSCItreefb5c8e44eb16b6044aa6049871d33106e1b6411atargetblankfb5c8e4a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/theandyb/LSCI/tree/fb5c8e44eb16b6044aa6049871d33106e1b6411a" target="_blank">fb5c8e4</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomtheandybLSCItreefb5c8e44eb16b6044aa6049871d33106e1b6411atargetblankfb5c8e4a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/theandyb/LSCI/tree/fb5c8e44eb16b6044aa6049871d33106e1b6411a" target="_blank">fb5c8e4</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/1000G_single_postion_comp_cache/
    Ignored:    analysis/disqus.html

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/outline.Rmd</code>) and HTML (<code>docs/outline.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/theandyb/LSCI/blob/fb5c8e44eb16b6044aa6049871d33106e1b6411a/analysis/outline.Rmd" target="_blank">fb5c8e4</a>
</td>
<td>
Andy Beck
</td>
<td>
2021-11-10
</td>
<td>
add BRIDGES comparison
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/theandyb/LSCI/96d087d9c30d273bf323cfca13af597ac36c49da/docs/outline.html" target="_blank">96d087d</a>
</td>
<td>
Andy Beck
</td>
<td>
2021-11-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/theandyb/LSCI/blob/752a4211afee694503eb52ce8970a1e447e3f8b7/analysis/outline.Rmd" target="_blank">752a421</a>
</td>
<td>
Andy Beck
</td>
<td>
2021-11-10
</td>
<td>
Add chisq to comp
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/theandyb/LSCI/79833a27d936abf8a4ad98362f85e47cc7326f48/docs/outline.html" target="_blank">79833a2</a>
</td>
<td>
Andy Beck
</td>
<td>
2021-11-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/theandyb/LSCI/blob/d8bd3c9c5753f3200a1be3d575af77357019a322/analysis/outline.Rmd" target="_blank">d8bd3c9</a>
</td>
<td>
Andy Beck
</td>
<td>
2021-11-10
</td>
<td>
update outline
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/theandyb/LSCI/bbebde65a26d23563ff5613cd63cb88292dbfae3/docs/outline.html" target="_blank">bbebde6</a>
</td>
<td>
Andy Beck
</td>
<td>
2021-11-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/theandyb/LSCI/blob/099f63d38cdb3fb858194aa3ef62a39fb214c126/analysis/outline.Rmd" target="_blank">099f63d</a>
</td>
<td>
Andy Beck
</td>
<td>
2021-11-10
</td>
<td>
update outline
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/theandyb/LSCI/896eb471513a85130f8e6618ca61f014fb8359d7/docs/outline.html" target="_blank">896eb47</a>
</td>
<td>
Andy Beck
</td>
<td>
2021-11-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/theandyb/LSCI/blob/0950949ab082423d1fe77f5cf2e218c094d48707/analysis/outline.Rmd" target="_blank">0950949</a>
</td>
<td>
Andy Beck
</td>
<td>
2021-11-09
</td>
<td>
wflow_publish(c(“analysis/1000G_single_postion_comp.Rmd”, “analysis/outline.Rmd”))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/theandyb/LSCI/a878a58d134536c00f6497b1f35f4ede5facba80/docs/outline.html" target="_blank">a878a58</a>
</td>
<td>
Andy Beck
</td>
<td>
2021-11-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/theandyb/LSCI/blob/53c69c2d5b7ecea86b700e1e0ef1f8ab44b8da7e/analysis/outline.Rmd" target="_blank">53c69c2</a>
</td>
<td>
Andy Beck
</td>
<td>
2021-11-09
</td>
<td>
Add some comparison analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/theandyb/LSCI/6922f0ab0ca503ddaada99e212e448b6da6076fa/docs/outline.html" target="_blank">6922f0a</a>
</td>
<td>
Andy Beck
</td>
<td>
2021-11-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/theandyb/LSCI/blob/51b4185eda9de328e9ce75bef63d13177ea80392/analysis/outline.Rmd" target="_blank">51b4185</a>
</td>
<td>
Andy Beck
</td>
<td>
2021-11-09
</td>
<td>
add disqus
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/theandyb/LSCI/c2bc42e367c2c9e6186bb69799a1f456102cda00/docs/outline.html" target="_blank">c2bc42e</a>
</td>
<td>
Andy Beck
</td>
<td>
2021-11-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/theandyb/LSCI/blob/e05bd2169b12f979f33c316eb5067add6f2c8f77/analysis/outline.Rmd" target="_blank">e05bd21</a>
</td>
<td>
Andy Beck
</td>
<td>
2021-11-09
</td>
<td>
outline as of 11/9/2021
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/theandyb/LSCI/eeb92e157b59ab5817dd7f7b6b075f38f3d48346/docs/outline.html" target="_blank">eeb92e1</a>
</td>
<td>
Andy Beck
</td>
<td>
2021-11-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/theandyb/LSCI/blob/98ee097aeaf10b656d878734a7b5047be9e57b95/analysis/outline.Rmd" target="_blank">98ee097</a>
</td>
<td>
Andy Beck
</td>
<td>
2021-11-04
</td>
<td>
cleanup
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/theandyb/LSCI/98ee097aeaf10b656d878734a7b5047be9e57b95/docs/outline.html" target="_blank">98ee097</a>
</td>
<td>
Andy Beck
</td>
<td>
2021-11-04
</td>
<td>
cleanup
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="paper-outline" class="section level1">
<h1>Paper Outline</h1>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>This is an outline of a prospective paper if we were to include results from both the 1000G deep sequencing data and the BRIDGES consortium GWAS. Both studies are of comparable sample size, but differ in that BRIDGES is entirely European ancestry whereas 1000G has 5 constituent super-populations (AFR, AMR, EAR, EUR, and SAS) which can be further distilled into more specific populations.</p>
<ul>
<li>Why interest in studying the germline mutation rate?
<ul>
<li>Source of heritable genetic variation -&gt; foundational importance in human genetics</li>
<li>Shapes the genetic diversity of populations</li>
<li>Insights into DNA damage / repair mechanisms</li>
<li>Uses in applications such as tests for selective sweeps (things that need a background null model)</li>
</ul></li>
<li>Why does it vary across the genome?
<ul>
<li>Variation at different scales (small to megabase patterns of variation)</li>
<li>Replication timing, chromatin accessibility, local sequence context</li>
</ul></li>
<li>How have we gone about studying the influence of local sequence context, and why is it challenging?
<ul>
<li><em>de novo</em> variants: pro = direct look; con = too few to do fine-grained analysis across the genome</li>
<li>Common non-coding variants: more plentiful, distribution influenced by gcBGC (and probably a little taste of selection)</li>
<li>Singleton variants: newer evolutionary -&gt; less influenced by selection, gcBGC; vastly more plentiful</li>
<li>7mer models &gt; 5mer &gt; 3mer</li>
<li>Number of unique motifs grows exponentially as context grows</li>
</ul></li>
<li>What are we doing here?
<ul>
<li>Detailed characterization of the (marginal) influence of nucleotides at flanking positions</li>
<li>Evaluation of the strength of two-position interactions</li>
<li>Evaluating consistency across datasets and between 1000G populations</li>
<li>Questions of interest
<ul>
<li>for modeling the mutation rate at the granularity of a single base, how much local context needs to be accounted for, and what order interactions are important?</li>
<li>do we see any evidence of differential mutation generation process across ancestry groups?</li>
</ul></li>
</ul></li>
<li>What do we show?
<ul>
<li>Influence of single positions strongest near focal site and decreases precipitously
<ul>
<li>Significant statistics at +/- 10 -&gt; noise? (how to delineate between signal and noise?)</li>
</ul></li>
<li>Interactions appear to play a role, but contained within +/- ….</li>
</ul></li>
</ul>
</div>
<div id="methods" class="section level2">
<h2>Methods</h2>
<div id="datasets" class="section level3">
<h3>Datasets</h3>
<div id="bridges" class="section level4">
<h4>BRIDGES</h4>
<p><em>Pull the details from Jed’s paper</em></p>
<ul>
<li>GRCh37</li>
</ul>
</div>
<div id="g" class="section level4">
<h4>1000G</h4>
<ul>
<li>2,504 unrelated individuals from the 1000G deep-sequence data</li>
<li>GRCh38</li>
</ul>
</div>
</div>
<div id="data-prep" class="section level3">
<h3>Data prep</h3>
<ul>
<li>VCFs filtered down to singletons</li>
<li>Each singleton annotated with simple mutation subtype, 21-mer motif centered at site
<ul>
<li>12 mutation sub-types collapsed into 6 by taking reverse complement</li>
<li>C/G &gt; NN sub-types split into CpG and non-CpG categories</li>
</ul></li>
<li>Each singleton matched to 5 nearby (321bp) control sites with matching REF allele</li>
<li>Background rates estimated genomewide
<ul>
<li>Counts of A, C, G, T stratified by nucleotides in +/- 10 bp window and by all pair of positions within that window</li>
</ul></li>
</ul>
</div>
<div id="single-position-models" class="section level3">
<h3>Single Position Models</h3>
<div id="chi-square-goodness-of-fit" class="section level4">
<h4>Chi Square Goodness of Fit</h4>
<p>For each of the 9 mutation sub-types, we stratify the counts of the singletons of that subtype by the nucleotides at each flanking position within the +/- 10 bp window centered at the singleton. For each relative position, we evaluate the influence of local context at that position by comparing the counts for each nucleotide to the distribution with rates estimated either genome-wide or based on the control sample for that sub-type. In the former, we posit that the nucleotide at a flanking position relative to a singleton of type XY&gt;JK should be a random draw from the observed genome-wide distribution of nucleotides flanking XY at the same relative position, while in the latter we posit that the draws were from a distribution matching the nucleotides flanking control sites sampled for each singleton. Deviations from these models suggest the occurrence of a singleton (mutation) are more or less likely when paricular nucleotides are observed at a flanking site. We use the chi-square goodness of fit statistic to evaluate the departure of our singleton observations from expectations based on the genome-wide background rate and the control sample distributions.</p>
</div>
</div>
<div id="two-position-models" class="section level3">
<h3>Two Position Models</h3>
<p>The “fuel” for these models is once again singleton/control/genome-wide conunts stratified by the nucleotides at flanking positions, but here we stratify into contingency tables based on the nucleotides at two flanking positions (within the +/- 10 bp window we have 190 such pairs to consider).</p>
<div id="chi-square-test-of-independence" class="section level4">
<h4>Chi Square Test of Independence</h4>
<ul>
<li>Q: are cell frequencies product of row and marginal?</li>
<li>Can be done for singletons, controls, and genome-wide</li>
</ul>
</div>
<div id="loglinear-model" class="section level4">
<h4>Loglinear Model</h4>
<p>Above test of independence is likely to be rejected due to known result that di-nucleotide distributions are not reflective of marginal distributions of individual nucleotides (i.e. frequency of AA is not square of the frequency of A genome-wide). When our task is ultimately to predict rates of mutation, the more pertinent question is whether or not there is a pattern of interaction that is unique to the singletons. Following the example of Zhu, et al (2017), we employ log-linear regression models to identify pairs of positions where the interactions between nucleotides in the singletons is different from the interaction we see in the control sample. We do this by fitting the following model for each pair of positions:</p>
<p><span class="math display">\[
\log{(n_{ijs})} = \lambda_0 + \lambda_i + \lambda_j + \lambda_s + \lambda_{ij} + \lambda_{is} + \lambda_{js}
\]</span></p>
<p>where <span class="math inline">\(n_{ijs}\)</span> is the observed count for the sub-type flanked by nucleotides i, j in status s (either singletons or controls).</p>
</div>
</div>
</div>
<div id="results" class="section level2">
<h2>Results</h2>
<div id="single-position-models-1" class="section level3">
<h3>Single Position Models</h3>
<ul>
<li><a href="BRIDGES_single_position.html">BRIDGES Results</a></li>
<li><a href="1000G_single_position_ALL.html">1000G Results</a></li>
<li><a href="1000G_single_postion_comp.html">1000G Cross-Population Comparison</a></li>
</ul>
<p>Across all sub-types, we see the strongest “signal” at positions more focal to center, with statistics dropping as we move further up/down stream from the site of interest. Due to the large number of singleton observations, we have very high power to detect even small deviations from the null model; this results in large statistics even at the +/- 10 bp position (albeit much smaller in magnitude relative to positions closer to the singleton site).</p>
<p class="text-danger">
Problems
</p>
<ul>
<li>Large counts result in high power to detect small differences in distribution
<ul>
<li>How to delineate between “statistically significant” and “biologically” significant here?</li>
</ul></li>
</ul>
<p class="text-success">
Ideas for further refinement
</p>
<ul>
<li>Reframe question in terms of a “classification problem”
<ul>
<li>I.e., given the nucleotide at a flanking position, how well are we able to distinguish between singletons and controls?</li>
</ul></li>
<li>Purposefully construct a null by stratifying singleton counts based on a nucleotide far enough up/down stream that this we would expect to be a random draw of some sorts(?)
<ul>
<li>Goal is to characterize how much “noise” we expect in the background</li>
</ul></li>
</ul>
<div id="consistency-across-1000g-super-populations" class="section level4">
<h4>Consistency across 1000G super-populations</h4>
<ul>
<li>Statistics highly correlated across position for each sub-type and pair of populations</li>
<li>Individual nucleotide contribution to each position’s statistic also highly correlated between populations</li>
<li>Chi-sq approach: use each super-population to fit counts for each other -&gt; statistics much smaller than when we use control distributions to fit same count tables</li>
</ul>
</div>
</div>
<div id="two-position-models-1" class="section level3">
<h3>Two-Position Models</h3>
<ul>
<li><p><a href="BRIDGES_two_positions.html">BRIDGES Results</a></p></li>
<li><p><a href="1000G_two_positions_ALL.html">1000G Results</a></p></li>
<li><p><a href="1000G_two_postions_comp.html">1000G Cross-Population Comparison</a></p></li>
<li><p>Chi-square tests of independence</p>
<ul>
<li>For each pair of positions, proportions of singletons stratified by “row” and “column” nucleotides are not products of row and column marginal distributions</li>
<li>Strongest deviation from independence assumption at pairs of positions directly adjacent to one another.</li>
<li>This is unsurprising since we see this both in the corresponding genome-wide and control distribution counts</li>
</ul></li>
<li><p>Log-linear models: is the interaction we see in singletons unique to singletons?</p>
<ul>
<li>Model: for each pair of positions, model the counts of both the singletons and the controls as a function of the nucleotides at the two positions</li>
<li>Model allows for different single position parameters for singletons and controls</li>
<li>Model also includes interaction term between the nucleotides at the two positions</li>
<li>Model does not include the three-way interaction term (which would fully saturate the model)</li>
<li>Deviance compares the fitted model to the fully saturated model -&gt; how different is the interaction in singletons to that if you assume it’s the same between singletons and controls?</li>
<li><p class="text-info">
Results
</p>
<ul>
<li>higher statistics at pairs of positions near singleton, with quick drop off as distance between relative positions and distance between relative positions and singleton increase</li>
</ul></li>
<li><p class="text-danger">
Issues
</p>
<ul>
<li>Same problem discussed above regarding the difference between statistical significance and practical significance.</li>
</ul></li>
</ul></li>
</ul>
</div>
</div>
<div id="discussion" class="section level2">
<h2>Discussion</h2>
</div>
</div>

<div id="disqus_thread"></div>
<script>
(function() { // DON'T EDIT BELOW THIS LINE
var d = document, s = d.createElement('script');
s.src = 'https://lsci-doc.disqus.com/embed.js';
s.setAttribute('data-timestamp', +new Date());
(d.head || d.body).appendChild(s);
})();
</script>
<noscript>Please enable JavaScript to view the
<a href="https://disqus.com/?ref_noscript">
  comments powered by Disqus.</a></noscript>

<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
