---
title: "1000G_single_position_ALL"
author: "Andy Beck"
date: "2021-11-01"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

```{r echo=FALSE, message=FALSE, warning=FALSE}
library(tidyverse)
library(tikzDevice)
width = 4
height = 4

source("code/single_pos_functions.R")
```


# 1000G Single Position Models - All 

## Introduction

Here in this document we present the results for the single position models using the singletons from all 2,504 unrelated individuals in the 1000G deep sequence data. Comparisons between the five super-populations can be found in the [comparison document](1000G_single_postion_comp.html). First we'll look at both the collapsed and un-collapsed categories to verify that we do not lose information when we take reverse-complements.

### Model Descriptions

#### Goodness of Fit Approach

For each singleton sub-type, we evaluate the "independent influence" of nucleotides at flanking positions up to 10 bases up or down-stream from the site of interest by modeling the observed count table

| Nucleotide | N |
|:----------:|:-:|
| A | $n_A$ |
| C | $n_C$ |
| G | $n_G$ |
| T | $n_T$ |

using rates we observe nucleotides flanking the same reference allele genome-wide or in a sample of control observations sampled for each singleton. That is, for each nucleotide we compute an expected count based on the frequency observed in either the genome-wide distribution or the control sample ( $ e_{X,i} = \hat{p}_{X,i} \times n_S$ ) and use the chi-square goodness of fit statistic to assess deviance from this null model (i.e. are singletons uniformly sampled from either the genome-wide or control distribution)?

$$
\chi_i = \sum_{x \in \{A,C,G,T\} }\frac{(n_{x,i} - e_{x,i})^2}{e_{x,i}}, i \in (-10,-1),(1,10) 
$$

#### Relative Entropy

Using the expectation under the control-distribution estimated rates, we can also compute the relative entropy between our observed singleton rates and the control rates.

$$
KL(S||C) = \sum_{i \in \{A,C,G,T\}} p_i \log(\frac{p_i}{q_i})
$$

As a measure of distance, we again have the interpretation of larger = bigger difference in distributions at a flanking position. Unlike the GoF approach above, here we don't have the problem of inflation of the statistic due to our large sample size.

### Log Linear Model Approach

Following the example of Zhu et al (2017), we also construct models of the form

$$
\log(n_{ij}) = \lambda_0 + \sum_{k \in \{C,G,T\}} \lambda_j I(i=k) + \lambda_{s}I(j=s)
$$
where we jointly model the counts for singletons and their matched controls stratified by the nucleotides at a given flanking position in the +/- 10 bp window. Note that here the model assumes a shared nucleotide distribution between the singletons and the controls; allowing for a difference through an interaction term would yield a fully saturated model. Zhu et al (2017) use the deviance statistic to identify putative mutation motifs, which compares the above reduced model to the fully saturated model, with a larger deviance statistic resulting from a larger difference between the observed distribution of nucleotides flanking singleton and control sites from the fitted distribution based on the reduced model. The deviance statistic tends to increase due to sample size, so Zhu et al (2017) additionally divided the deviance statistic by twice the sample size to yield the relative entropy/KL divergence, which is not sensitive to sample size and allowed for comparisons between mutation sub types. We choose to take a similar approach in that we compute the KL divergence between the observed and fitted frequencies for only the singletons.

## 9 Subtype Results

The nine basic subtypes are AT_CG, AT_GC, AT_TA, GC_AT, GC_TA, GC_CG, cpg_GC_AT, cpg_GC_TA, and cpg_GC_CG, where the notation XY_JK indicates we either observed a X>J or Y>K substitution, with XY and JK being complementary to each other. For each sub-type with reference alleles XY and position i within a 21-mer window centered at the singleton, we model the distributions of nucleotides A, C, G and T flanking the singletons using the rates we observe either genome-wide flanking the reference nucleotides XY at the same relative position or the rates of nucleotides flanking control positions sampled for each singleton. We use the chi-square goodness of fit statistic to "measure" the degree to which this model doesn't hold. First, let's plot the deviance statistics we observe at each relative position for each sub-type.

### Singletons per Subtype

|   | ALL | AFR | AMR | EAS | EUR | SAS |
|:-:|:---:|:---:|:---:|:---:|:---:|:---:|
| AT_CG | 3994644 | 1160586 | 490617 | 871253 | 653113 | 819915 |
| AT_GC | 13689402 | 3958441 | 1735039 | 3070427 | 2207497 | 2719543 |
| AT_TA | 3620603 | 1032682 | 447998 | 828306 | 586770 | 725607 |
| GC_AT | 12388995| 3583196 | 1572949 | 2721559 | 1977163 | 2535890 |
| GC_CG | 4246208 | 1253167 | 528073 | 939800 | 684128 | 841931 |
| GC_TA | 4576214 | 1327907 | 568874 | 1013793 | 743282 | 923166 |
| cpg_GC_AT | 5990034 | 1749306 | 748097 | 1315852 | 965863 | 1211353 |
| cpg_GC_CG | 389681 | 109897 | 49895 | 86088 | 62986 | 80857 |
| cpg_GC_TA | 492238 | 142803 | 59752 | 109941 | 77637 | 102155 |

```{r, echo=FALSE}
cpg_stat_by_pos <- function(sp, data_dir = "/net/snowwhite/home/beckandy/research/1000G_LSCI/output/single_pos_df/"){
  df <- load_all_sp_results(sp, "all_GC_AT", data_dir = data_dir)
  df <- df %>%
    select(rp, deviance, re) %>%
    group_by(rp) %>%
    summarize(deviance = sum(deviance),
              re = sum(re))
  return(df)
}

```

### Demonstrate CpG effect

```{r}
cpg_stat_by_pos("ALL") %>%
  ggplot(aes(x = rp, y = re)) +
  geom_point() + 
  geom_line() +
  ggtitle("C â†’ T: CpG Effect") +
  xlab("Relative Position") +
  ylab("Relative Entropy")
```


### AT_CG

```{r echo=FALSE, message=FALSE}
st <- "AT_CG"
p1 <- plot_pos_stat("ALL", st, zhu_by_pos, "Relative Entropy by Position: ", ylab_text = "Relative Entropy")
p2 <- plot_nuc_level_bar("ALL", st, "re", "Nucleotide Contribution to Overall Statistic",
                         paste0("Subtype: ", st, "; Population: All"))
p3 <- plot_signed_re_by_pos("ALL", st)

p1
p2
p3
```

```{r results='asis', echo=FALSE}
gof_by_position("ALL", st) %>% knitr::kable()
```

Here we see that the  strongest influence (or deviation from expectation) occurs at the -1 relative position, and in general decreases as we move further from the focal site. Although the statistics do drop drastically as we move towards +/- 10, the unadjusted statistics are still statistically significant (chi-square null distribution with 4 - 1 = 3 degrees of freedom).

### AT_GC

```{r echo=FALSE, message=FALSE}
st <- "AT_GC"
p1 <- plot_pos_stat("ALL", st, re_by_position, "Relative Entropy by Position: ", ylab_text = "Relative Entropy")
p2 <- plot_nuc_level_bar("ALL", st, "chi_sq_ct", "Nucleotide Contribution to Overall Statistic",
                         paste0("Subtype: ", st, "; Population: All"))
p3 <- plot_signed_re_by_pos("ALL", st)

p1
p2
p3
```


Here for the AT_GC transition we see a similar pattern to what we observed above, but with the most influential marginal effect occurring at the +1 position. 

### AT_TA

```{r echo=FALSE, message=FALSE}
st <- "AT_TA"
p1 <- plot_pos_stat("ALL", st, re_by_position, "Relative Entropy by Position: ", ylab_text = "Relative Entropy")
p2 <- plot_nuc_level_bar("ALL", st, "chi_sq_ct", "Nucleotide Contribution to Overall Statistic",
                         paste0("Subtype: ", st, "; Population: All"))
p3 <- plot_signed_re_by_pos("ALL", st)

p1
p2
p3
```

Here for the A>T(T>A) transversion we see that the signal at the +1 is far greater than what we observe at other relative positions. This is in contrast to what we observed above where marginal signals at multiple relative positions (generally the +/- 1 positions) stood out. Otherwise we again see the same trend of statistics drastically decreasing as we move further up or down stream. 

### GC_AT

```{r echo=FALSE, message=FALSE}
st <- "GC_AT"
p1 <- plot_pos_stat("ALL", st, re_by_position, "Relative Entropy by Position: ", ylab_text = "Relative Entropy")
p2 <- plot_nuc_level_bar("ALL", st, "chi_sq_ct", "Nucleotide Contribution to Overall Statistic",
                         paste0("Subtype: ", st, "; Population: All"))
p3 <- plot_signed_re_by_pos("ALL", st)

p1
p2
p3
```

Here we look at the G>A (C>T) transition, conditioning on sites not being CpGs (note: due to this the degrees of freedom for the test statistic at the +1 position is 2 instead of 3). Here we see that, with the exception of the -1 position, the control distribution-based statistic is greater than what we observe with the genome-wide-based statistic.  

### cpg_GC_AT

```{r echo=FALSE, message=FALSE}
st <- "cpg_GC_AT"
p1 <- plot_pos_stat("ALL", st, re_by_position, "Relative Entropy by Position: ", ylab_text = "Relative Entropy", r_start = 2)
p2 <- plot_nuc_level_bar("ALL", st, "chi_sq_ct", "Nucleotide Contribution to Overall Statistic",
                         paste0("Subtype: ", st, "; Population: All"), r_start =2)
p3 <- plot_signed_re_by_pos("ALL", st, r_start = 2)

p1
p2
p3
```

Here we condition on the +1 being a G for the G>A(C>T) transition. It is interesting to see for this sub-type that when we condition on the +1 being a G, we see that the control-based statistic behaves like it does for the other sub-types in that it drops precipitously as we move further up/down stream, but the genome-wide rate appears to plateau at a much higher value. It should also be noted that this is the only sub-type in which this pattern is observed.

### GC_TA

```{r echo=FALSE, message=FALSE}
st <- "GC_TA"
p1 <- plot_pos_stat("ALL", st, re_by_position, "Relative Entropy by Position: ", ylab_text = "Relative Entropy")
p2 <- plot_nuc_level_bar("ALL", st, "chi_sq_ct", "Nucleotide Contribution to Overall Statistic",
                         paste0("Subtype: ", st, "; Population: All"))
p3 <- plot_signed_re_by_pos("ALL", st)

p1
p2
p3
```


For the G>T(C>A) transversion at non-CpG sites we see that the largest marginal effect for the control-rate models appears at the +1 position, with the -1 position also appearing to be influential as well.

### cpg_GC_TA

```{r echo=FALSE, message=FALSE}
st <- "cpg_GC_TA"
p1 <- plot_pos_stat("ALL", st, re_by_position, "Relative Entropy by Position: ", ylab_text = "Relative Entropy", r_start = 2)
p2 <- plot_nuc_level_bar("ALL", st, "chi_sq_ct", "Nucleotide Contribution to Overall Statistic",
                         paste0("Subtype: ", st, "; Population: All"), r_start =2)
p3 <- plot_signed_re_by_pos("ALL", st, r_start = 2)

p1
p2
p3
```

When we condition on the site being a CpG, we see that the largest marginal influence appears at the -1 site, with a modest signal at the -4 and -5 positions as well.

### GC_CG

```{r echo=FALSE, message=FALSE}
st <- "GC_CG"
p1 <- plot_pos_stat("ALL", st, re_by_position, "Relative Entropy by Position: ", ylab_text = "Relative Entropy")
p2 <- plot_nuc_level_bar("ALL", st, "chi_sq_ct", "Nucleotide Contribution to Overall Statistic",
                         paste0("Subtype: ", st, "; Population: All"))
p3 <- plot_signed_re_by_pos("ALL", st)

p1
p2
p3
```

For the G<->C transversion at non-CpG sites we observe the largest marginal signal occurring at the -1 position.

### cpg_GC_CG

```{r echo=FALSE, message=FALSE}
st <- "cpg_GC_CG"
p1 <- plot_pos_stat("ALL", st, re_by_position, "Relative Entropy by Position: ", ylab_text = "Relative Entropy", r_start = 2)
p2 <- plot_nuc_level_bar("ALL", st, "chi_sq_ct", "Nucleotide Contribution to Overall Statistic",
                         paste0("Subtype: ", st, "; Population: All"), r_start =2)
p3 <- plot_signed_re_by_pos("ALL", st, r_start = 2)

p1
p2
p3
```

When we condition on the C<->G transversion being at a CpG site, we notice that the strongest marginal effect appears at the +2 position for both models, with sharp declines as we move away from +/- 3 bp.


## 18 Subtype Results

In the above analyses, we reduced the number of sub-types by taking the reverse complement of singletons whose reference alleles were either G or T. That is, for example, the AT_CG analysis at relative position -1 combined the counts of nucleotides observed -1 of A>C plus the counts of the complements of nucleotides observed +1 of T>G singletons. We do this because we don't generally (ever?) observe which strand the mutation originated on. Here we'll compare complementary "simple" sub-types to verify that we see the same patterns in both.

```{r echo=FALSE}
simple_rc <- function(nuc){
  map_vec <- c("A","C","G","T")
  names(map_vec) <- c("T","G","C","A")
  return(map_vec[nuc])
}

load_simple_results <- function(subtype, rp, sp){
  data_dir <- "/net/snowwhite/home/beckandy/research/1000G_LSCI/output/single_pos_df/"
  f_name <- paste0(data_dir, sp, "/6cat/", subtype, "_rp", rp, ".csv")
  df <- read_csv(f_name, col_types = cols())
  return(df)
}

load_simple_results_rc <- function(subtype, rp, sp){
  data_dir <- "/net/snowwhite/home/beckandy/research/1000G_LSCI/output/single_pos_df/"
  f_name <- paste0(data_dir, sp, "/6cat/", subtype, "_rp", (-1*rp), ".csv")
  df <- read_csv(f_name, col_types = cols()) %>%
    rowwise() %>%
    mutate(Nuc = simple_rc(Nuc))
  return(df)
}

load_all_simple_results <- function(subtype, sp, r_start = 1){
  df <- load_simple_results(subtype, -10, sp)
  df$rp <- -10
  for(i in c(-9:-1, r_start:10)){
    df2 <- load_simple_results(subtype, i, sp)
    df2$rp <- i
    df <- bind_rows(df, df2)
  }
  return(df)
}

load_all_simple_results_rc <- function(subtype, sp, r_start = 1){
  df <- load_simple_results_rc(subtype, -10, sp)
  df$rp <- -10
  for(i in c(-9:-1, r_start:10)){
    df2 <- load_simple_results_rc(subtype, i, sp)
    df2$rp <- i
    df <- bind_rows(df, df2)
  }
  return(df)
}

simple_gof_by_position <- function(subtype, sp, r_start = 1, rc = FALSE){
  final <- data.frame(pos = numeric(), chi_sq = numeric(), type = character())
  for(i in c(-10:-1, r_start:10)){
    if(rc){
      df <- load_simple_results_rc(subtype, i, sp)
    } else {
      df <- load_simple_results(subtype, i, sp)
    }
    final <- bind_rows(final,
                       data.frame(pos = c(i,i), 
                                  chi_sq = c(sum(df$chi_sq_gw, na.rm = T), sum(df$chi_sq_ct, na.rm = T) ), 
                                  type = c("Genome-wide", "Control")))
  }
  return(final)
}

plot_simple_pos_stat <- function(subtype, sp, r_start = 1, rc = FALSE){
  df <- simple_gof_by_position(subtype, sp, r_start, rc)
  p <-df %>%
    ggplot(aes(x = pos, y = chi_sq, colour = type)) +
    geom_point() +
    geom_line() +
    ggtitle(paste0("1000G Single-Position Results: ", subtype,
                   ifelse(rc, "(rc)", "")),
            paste0("Population: ", sp)) +
    xlab("Relative Position") +
    ylab("Chi Square Goodness of Fit Statistic") +
    labs(colour = "Background Rate")
  return(p)
}
```

### AT_CG

For our first analysis, let's compare the results we observe in the AT_CG sub-type to what we observe for the two simple sub-types which contribute to this composite sub-type: A>C and T>G. Note that in order to compare T>G to AT>CG we'll need to take reverse complements, i.e. results for relative position 1 for T>G "flipped" to -1, 2 to -2, etc.

```{r echo=FALSE}
plot_simple_pos_stat("A_C", "ALL")
plot_simple_pos_stat("T_G", "ALL", rc = TRUE)
```

In the above two plots we don't see any differences in the patterns for the A_C and T_G subtypes, suggesting that collapsing the two into one category is acceptable. 

### AT_GC

```{r echo=FALSE}
plot_simple_pos_stat("A_G", "ALL")
plot_simple_pos_stat("T_C", "ALL", rc = TRUE)
```

### AT_TA

```{r echo=FALSE}
plot_simple_pos_stat("A_T", "ALL")
plot_simple_pos_stat("T_A", "ALL", rc = TRUE)
```

### GC_AT

```{r echo=FALSE}
plot_simple_pos_stat("C_T", "ALL")
plot_simple_pos_stat("G_A", "ALL", rc = TRUE)
```

### cpg_GC_AT

```{r echo=FALSE}
plot_simple_pos_stat("cpg_C_T", "ALL", r_start = 2)
plot_simple_pos_stat("cpg_G_A", "ALL", rc = TRUE, r_start = 2)
```

### GC_TA

```{r echo=FALSE}
plot_simple_pos_stat("C_A", "ALL")
plot_simple_pos_stat("G_T", "ALL", rc = TRUE)
```

### cpg_GC_AT

```{r echo=FALSE}
plot_simple_pos_stat("cpg_C_A", "ALL", r_start = 2)
plot_simple_pos_stat("cpg_G_T", "ALL", rc = TRUE, r_start = 2)
```

### GC_CG

```{r echo=FALSE}
plot_simple_pos_stat("C_G", "ALL")
plot_simple_pos_stat("G_C", "ALL", rc = TRUE)
```

### cpg_GC_AT

```{r echo=FALSE}
plot_simple_pos_stat("cpg_C_G", "ALL", r_start = 2)
plot_simple_pos_stat("cpg_G_C", "ALL", rc = TRUE, r_start = 2)
```

I do not see any substantial differences between simple sub-type pairs, and so forth I will only consider the 9 collapsed categories.

