---
title: "BRIDGES: Single Position Models"
author: "Andy Beck"
date: "2023-02-06"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

```{r echo=FALSE, message=FALSE, warning=FALSE}
library(tidyverse)
library(tikzDevice)
library(sjPlot)
width = 4
height = 4

source("code/single_pos_functions.R")
## Plot options
theme_set(theme_bw())
plot_w <- 800
plot_h <- 500
plot_dpi <- 300

cbPalette <- c("#999999", "#E69F00", "#56B4E9", "#009E73", "#F0E442", "#0072B2", "#D55E00", "#CC79A7")
pic_dir <- "/net/snowwhite/home/beckandy/research/BRIDGES_redo/output/figs/single_pos/"

df_dir <- "/net/snowwhite/home/beckandy/research/BRIDGES_redo/output/single_pos_df/"
```


# BRIDGES Single Position Models

## Introduction

Here in this document we present the results for the single position models using the singletons from the BRIDGES Consortium data. For this analysis we will only consider the collapsed mutation subtype categories.

### Model Descriptions

#### Goodness of Fit Approach

For each singleton sub-type, we evaluate the "independent influence" of nucleotides at flanking positions up to 10 bases up or down-stream from the site of interest by modeling the observed count table

| Nucleotide | N |
|:----------:|:-:|
| A | $n_A$ |
| C | $n_C$ |
| G | $n_G$ |
| T | $n_T$ |

using rates we observe nucleotides flanking the same reference allele genome-wide or in a sample of control observations sampled for each singleton. That is, for each nucleotide we compute an expected count based on the frequency observed in either the genome-wide distribution or the control sample ( $ e_{X,i} = \hat{p}_{X,i} \times n_S$ ) and use the chi-square goodness of fit statistic to assess deviance from this null model (i.e. are singletons uniformly sampled from either the genome-wide or control distribution)?

$$
\chi_i = \sum_{x \in \{A,C,G,T\} }\frac{(n_{x,i} - e_{x,i})^2}{e_{x,i}}, i \in (-10,-1),(1,10) 
$$

#### Relative Entropy

Using the expectation under the control-distribution estimated rates, we can also compute the relative entropy between our observed singleton rates and the control rates.

$$
KL(S||C) = \sum_{i \in \{A,C,G,T\}} p_i \log(\frac{p_i}{q_i})
$$

As a measure of distance, we again have the interpretation of larger = bigger difference in distributions at a flanking position. Unlike the GoF approach above, here we don't have the problem of inflation of the statistic due to our large sample size.

### Log Linear Model Approach

Following the example of Zhu et al (2017), we also construct models of the form

$$
\log(n_{ij}) = \lambda_0 + \sum_{k \in \{C,G,T\}} \lambda_j I(i=k) + \lambda_{s}I(j=s)
$$
where we jointly model the counts for singletons and their matched controls stratified by the nucleotides at a given flanking position in the +/- 10 bp window. Note that here the model assumes a shared nucleotide distribution between the singletons and the controls; allowing for a difference through an interaction term would yield a fully saturated model. Zhu et al (2017) use the deviance statistic to identify putative mutation motifs, which compares the above reduced model to the fully saturated model, with a larger deviance statistic resulting from a larger difference between the observed distribution of nucleotides flanking singleton and control sites from the fitted distribution based on the reduced model. The deviance statistic tends to increase due to sample size, so Zhu et al (2017) additionally divided the deviance statistic by twice the sample size to yield the relative entropy/KL divergence, which is not sensitive to sample size and allowed for comparisons between mutation sub types. We choose to take a similar approach in that we compute the KL divergence between the observed and fitted frequencies for only the singletons.


## Results

### Recapitulate CpG Effect

```{r, echo=FALSE}
cpg_stat_by_pos <- function(sp, data_dir){
  df <- load_all_sp_results(sp, "all_GC_AT", data_dir = data_dir)
  df <- df %>%
    select(rp, deviance, re) %>%
    group_by(rp) %>%
    summarize(deviance = sum(deviance),
              re = sum(re))
  return(df)
}

```

We first check to make sure that nothing has gone catastrophically wrong by ensuring that we see the known CpG effect for the C -> T subtype. 

```{r}
cpg_stat_by_pos("BRIDGES", df_dir) %>%
  ggplot(aes(x = rp, y = re)) +
  geom_point() + 
  geom_line() +
  ggtitle("C â†’ T: CpG Effect") +
  xlab("Relative Position") +
  ylab("Relative Entropy") + scale_color_manual(values = cbPalette)

f_out <- paste0(pic_dir, "ALL_CpG_effect.png")
ggsave(f_out, width = 800, height = 500, units = "px", dpi = 120)

plot_signed_re_by_pos("ALL","all_GC_AT") + scale_color_manual(values = cbPalette)
f_out <- paste0(pic_dir, "ALL_CpG_res.png")
ggsave(f_out, width = 800, height = 500, units = "px", dpi = 120)
```

Based on the above two figures, we see the strongest influence of nucleotides at an individual position at the +1 position, and the primary driver of this statistic is an enrichment for G at this position, matching what we'd expect.

### AT_CG

```{r echo=FALSE, message=FALSE}
st <- "AT_CG"
p1 <- plot_pos_stat("BRIDGES", st, zhu_by_pos, 
                    "Relative Entropy by Position: ", ylab_text = "Relative Entropy", data_dir = df_dir) +
  scale_color_manual(values = cbPalette)
p2 <- plot_nuc_level_bar("BRIDGES", st, "re", "Nucleotide Contribution to Overall Statistic", data_dir = df_dir)
p3 <- plot_signed_re_by_pos("BRIDGES", st, data_dir = df_dir) 

p1
f_out <- paste0(pic_dir, st, "_pos_stat.png")
ggsave(f_out, width = 800, height = 500, units = "px", dpi = 120)

p2 + scale_fill_manual(values = cbPalette)
f_out <- paste0(pic_dir, st, "_nuc_bar.png")
ggsave(f_out, width = 800, height = 500, units = "px", dpi = 120)

p3 + scale_color_manual(values = cbPalette)
f_out <- paste0(pic_dir, st, "_sign_re.png")
ggsave(f_out, width = 800, height = 500, units = "px", dpi = 120)
```

```{r results='asis', echo=FALSE}
zhu_by_pos("BRIDGES", st, data_dir = df_dir) %>% knitr::kable()
```

Here we see that the  strongest influence (or deviation from expectation) occurs at the -1 relative position, and in general decreases as we move further from the focal site. Although the statistics do drop drastically as we move towards +/- 10, the unadjusted statistics are still statistically significant (chi-square null distribution with 4 - 1 = 3 degrees of freedom).

### AT_GC

```{r echo=FALSE, message=FALSE}
st <- "AT_GC"
p1 <- plot_pos_stat("BRIDGES", st, zhu_by_pos, 
                    "Relative Entropy by Position: ", ylab_text = "Relative Entropy", data_dir = df_dir) +
  scale_color_manual(values = cbPalette)
p2 <- plot_nuc_level_bar("BRIDGES", st, "re", "Nucleotide Contribution to Overall Statistic", data_dir = df_dir)
p3 <- plot_signed_re_by_pos("BRIDGES", st, data_dir = df_dir) 

p1
f_out <- paste0(pic_dir, st, "_pos_stat.png")
ggsave(f_out, width = 800, height = 500, units = "px", dpi = 120)

p2 + scale_fill_manual(values = cbPalette)
f_out <- paste0(pic_dir, st, "_nuc_bar.png")
ggsave(f_out, width = 800, height = 500, units = "px", dpi = 120)

p3 + scale_color_manual(values = cbPalette)
f_out <- paste0(pic_dir, st, "_sign_re.png")
ggsave(f_out, width = 800, height = 500, units = "px", dpi = 120)
```

```{r results='asis', echo=FALSE}
zhu_by_pos("BRIDGES", st, data_dir = df_dir) %>% knitr::kable()
```

Here for the AT_GC transition we see a similar pattern to what we observed above, but with the most influential marginal effect occurring at the +1 position. 

### AT_TA

```{r echo=FALSE, message=FALSE}
st <- "AT_TA"
p1 <- plot_pos_stat("BRIDGES", st, zhu_by_pos, 
                    "Relative Entropy by Position: ", ylab_text = "Relative Entropy", data_dir = df_dir) +
  scale_color_manual(values = cbPalette)
p2 <- plot_nuc_level_bar("BRIDGES", st, "re", "Nucleotide Contribution to Overall Statistic", data_dir = df_dir)
p3 <- plot_signed_re_by_pos("BRIDGES", st, data_dir = df_dir) 

p1
f_out <- paste0(pic_dir, st, "_pos_stat.png")
ggsave(f_out, width = 800, height = 500, units = "px", dpi = 120)

p2 + scale_fill_manual(values = cbPalette)
f_out <- paste0(pic_dir, st, "_nuc_bar.png")
ggsave(f_out, width = 800, height = 500, units = "px", dpi = 120)

p3 + scale_color_manual(values = cbPalette)
f_out <- paste0(pic_dir, st, "_sign_re.png")
ggsave(f_out, width = 800, height = 500, units = "px", dpi = 120)
```

```{r results='asis', echo=FALSE}
zhu_by_pos("BRIDGES", st, data_dir = df_dir) %>% knitr::kable()
```

Here for the A>T(T>A) transversion we see that the signal at the +1 is far greater than what we observe at other relative positions. This is in contrast to what we observed above where marginal signals at multiple relative positions (generally the +/- 1 positions) stood out. Otherwise we again see the same trend of statistics drastically decreasing as we move further up or down stream. 

### GC_AT

```{r echo=FALSE, message=FALSE}
st <- "GC_AT"
p1 <- plot_pos_stat("BRIDGES", st, zhu_by_pos, 
                    "Relative Entropy by Position: ", ylab_text = "Relative Entropy", data_dir = df_dir) +
  scale_color_manual(values = cbPalette)
p2 <- plot_nuc_level_bar("BRIDGES", st, "re", "Nucleotide Contribution to Overall Statistic", data_dir = df_dir)
p3 <- plot_signed_re_by_pos("BRIDGES", st, data_dir = df_dir) 

p1
f_out <- paste0(pic_dir, st, "_pos_stat.png")
ggsave(f_out, width = 800, height = 500, units = "px", dpi = 120)

p2 + scale_fill_manual(values = cbPalette)
f_out <- paste0(pic_dir, st, "_nuc_bar.png")
ggsave(f_out, width = 800, height = 500, units = "px", dpi = 120)

p3 + scale_color_manual(values = cbPalette)
f_out <- paste0(pic_dir, st, "_sign_re.png")
ggsave(f_out, width = 800, height = 500, units = "px", dpi = 120)
```

```{r results='asis', echo=FALSE}
zhu_by_pos("BRIDGES", st, data_dir = df_dir) %>% knitr::kable()
```

Here we look at the G>A (C>T) transition, conditioning on sites not being CpGs (note: due to this the degrees of freedom for the test statistic at the +1 position is 2 instead of 3). Here we see that, with the exception of the -1 position, the control distribution-based statistic is greater than what we observe with the genome-wide-based statistic.  

### all_GC_AT

```{r echo=FALSE, message=FALSE}
st <- "all_GC_AT"
p1 <- plot_pos_stat("BRIDGES", st, zhu_by_pos, 
                    "Relative Entropy by Position: ", ylab_text = "Relative Entropy", data_dir = df_dir) +
  scale_color_manual(values = cbPalette)
p2 <- plot_nuc_level_bar("BRIDGES", st, "re", "Nucleotide Contribution to Overall Statistic", data_dir = df_dir)
p3 <- plot_signed_re_by_pos("BRIDGES", st, data_dir = df_dir) 

p1
f_out <- paste0(pic_dir, st, "_pos_stat.png")
ggsave(f_out, width = 800, height = 500, units = "px", dpi = 120)

p2 + scale_fill_manual(values = cbPalette)
f_out <- paste0(pic_dir, st, "_nuc_bar.png")
ggsave(f_out, width = 800, height = 500, units = "px", dpi = 120)

p3 + scale_color_manual(values = cbPalette)
f_out <- paste0(pic_dir, st, "_sign_re.png")
ggsave(f_out, width = 800, height = 500, units = "px", dpi = 120)
```

```{r results='asis', echo=FALSE}
zhu_by_pos("BRIDGES", st, data_dir = df_dir) %>% knitr::kable()
```

### cpg_GC_AT

```{r echo=FALSE, message=FALSE}
st <- "cpg_GC_AT"
p1 <- plot_pos_stat("BRIDGES", st, zhu_by_pos, "Relative Entropy by Position: ", 
                    ylab_text = "Relative Entropy", r_start = 2, data_dir = df_dir)
p2 <- plot_nuc_level_bar("BRIDGES", st, "re", "Nucleotide Contribution to Overall Statistic", 
                         r_start = 2, data_dir = df_dir)
p3 <- plot_signed_re_by_pos("BRIDGES", st, r_start = 2, data_dir = df_dir)

p1
f_out <- paste0(pic_dir, st, "_pos_stat.png")
ggsave(f_out, width = 800, height = 500, units = "px", dpi = 120)

p2 + scale_fill_manual(values = cbPalette)
f_out <- paste0(pic_dir, st, "_nuc_bar.png")
ggsave(f_out, width = 800, height = 500, units = "px", dpi = 120)

p3 + scale_color_manual(values = cbPalette)
f_out <- paste0(pic_dir, st, "_sign_re.png")
ggsave(f_out, width = 800, height = 500, units = "px", dpi = 120)
```

```{r results='asis', echo=FALSE}
zhu_by_pos("BRIDGES", st, r_start = 2, data_dir = df_dir) %>% knitr::kable()
```

Here we condition on the +1 being a G for the G>A(C>T) transition. 

### GC_TA

```{r echo=FALSE, message=FALSE}
st <- "GC_TA"
p1 <- plot_pos_stat("BRIDGES", st, zhu_by_pos, 
                    "Relative Entropy by Position: ", ylab_text = "Relative Entropy", data_dir = df_dir) +
  scale_color_manual(values = cbPalette)
p2 <- plot_nuc_level_bar("BRIDGES", st, "re", "Nucleotide Contribution to Overall Statistic", data_dir = df_dir)
p3 <- plot_signed_re_by_pos("BRIDGES", st, data_dir = df_dir) 

p1
f_out <- paste0(pic_dir, st, "_pos_stat.png")
ggsave(f_out, width = 800, height = 500, units = "px", dpi = 120)

p2 + scale_fill_manual(values = cbPalette)
f_out <- paste0(pic_dir, st, "_nuc_bar.png")
ggsave(f_out, width = 800, height = 500, units = "px", dpi = 120)

p3 + scale_color_manual(values = cbPalette)
f_out <- paste0(pic_dir, st, "_sign_re.png")
ggsave(f_out, width = 800, height = 500, units = "px", dpi = 120)
```

```{r results='asis', echo=FALSE}
zhu_by_pos("BRIDGES", st, data_dir = df_dir) %>% knitr::kable()
```

For the G>T(C>A) transversion at non-CpG sites we see that the largest marginal effect for the control-rate models appears at the +1 position, with the -1 position also appearing to be influential as well.

### cpg_GC_TA

```{r echo=FALSE, message=FALSE}
st <- "cpg_GC_TA"
p1 <- plot_pos_stat("BRIDGES", st, zhu_by_pos, 
                    "Relative Entropy by Position: ", ylab_text = "Relative Entropy", data_dir = df_dir) +
  scale_color_manual(values = cbPalette)
p2 <- plot_nuc_level_bar("BRIDGES", st, "re", "Nucleotide Contribution to Overall Statistic", data_dir = df_dir)
p3 <- plot_signed_re_by_pos("BRIDGES", st, data_dir = df_dir) 

p1
f_out <- paste0(pic_dir, st, "_pos_stat.png")
ggsave(f_out, width = 800, height = 500, units = "px", dpi = 120)

p2 + scale_fill_manual(values = cbPalette)
f_out <- paste0(pic_dir, st, "_nuc_bar.png")
ggsave(f_out, width = 800, height = 500, units = "px", dpi = 120)

p3 + scale_color_manual(values = cbPalette)
f_out <- paste0(pic_dir, st, "_sign_re.png")
ggsave(f_out, width = 800, height = 500, units = "px", dpi = 120)
```

```{r results='asis', echo=FALSE}
zhu_by_pos("BRIDGES", st, r_start = 2, data_dir = df_dir) %>% knitr::kable()
```

When we condition on the site being a CpG, we see that the largest marginal influence appears at the -1 site, with a modest signal at the -4 and -5 positions as well.

### GC_CG

```{r echo=FALSE, message=FALSE}
st <- "GC_CG"
p1 <- plot_pos_stat("BRIDGES", st, zhu_by_pos, 
                    "Relative Entropy by Position: ", ylab_text = "Relative Entropy", data_dir = df_dir) +
  scale_color_manual(values = cbPalette)
p2 <- plot_nuc_level_bar("BRIDGES", st, "re", "Nucleotide Contribution to Overall Statistic", data_dir = df_dir)
p3 <- plot_signed_re_by_pos("BRIDGES", st, data_dir = df_dir) 

p1
f_out <- paste0(pic_dir, st, "_pos_stat.png")
ggsave(f_out, width = 800, height = 500, units = "px", dpi = 120)

p2 + scale_fill_manual(values = cbPalette)
f_out <- paste0(pic_dir, st, "_nuc_bar.png")
ggsave(f_out, width = 800, height = 500, units = "px", dpi = 120)

p3 + scale_color_manual(values = cbPalette)
f_out <- paste0(pic_dir, st, "_sign_re.png")
ggsave(f_out, width = 800, height = 500, units = "px", dpi = 120)
```

```{r results='asis', echo=FALSE}
zhu_by_pos("BRIDGES", st, data_dir = df_dir) %>% knitr::kable()
```

For the G<->C transversion at non-CpG sites we observe the largest marginal signal occurring at the -1 position.

### cpg_GC_CG

```{r echo=FALSE, message=FALSE}
st <- "cpg_GC_CG"
p1 <- plot_pos_stat("BRIDGES", st, zhu_by_pos, 
                    "Relative Entropy by Position: ", ylab_text = "Relative Entropy", data_dir = df_dir) +
  scale_color_manual(values = cbPalette)
p2 <- plot_nuc_level_bar("BRIDGES", st, "re", "Nucleotide Contribution to Overall Statistic", data_dir = df_dir)
p3 <- plot_signed_re_by_pos("BRIDGES", st, data_dir = df_dir) 

p1
f_out <- paste0(pic_dir, st, "_pos_stat.png")
ggsave(f_out, width = 800, height = 500, units = "px", dpi = 120)

p2 + scale_fill_manual(values = cbPalette)
f_out <- paste0(pic_dir, st, "_nuc_bar.png")
ggsave(f_out, width = 800, height = 500, units = "px", dpi = 120)

p3 + scale_color_manual(values = cbPalette)
f_out <- paste0(pic_dir, st, "_sign_re.png")
ggsave(f_out, width = 800, height = 500, units = "px", dpi = 120)
```

```{r results='asis', echo=FALSE}
zhu_by_pos("BRIDGES", st, r_start = 2, data_dir = df_dir) %>% knitr::kable()
```

When we condition on the C<->G transversion being at a CpG site, we notice that the strongest marginal effect appears at the +2 position for both models, with sharp declines as we move away from +/- 3 bp.

